// @sberdevices/plasma-web/components/Progress/Progress.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Progress = void 0;

var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));

var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));

var _plasmaCore = /*#__PURE__*/require("@sberdevices/plasma-core");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var statuses = {
  success: /*#__PURE__*/(0, _styledComponents.css)(["background-color:#09a552;"]),
  warning: /*#__PURE__*/(0, _styledComponents.css)(["background-color:#ee6820;"]),
  error: /*#__PURE__*/(0, _styledComponents.css)(["background-color:#df2638;"])
};

var StyledRoot = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web__sc-1e491uq-0"
})(["display:flex;align-items:center;"]);

var StyledTrack = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web__sc-1e491uq-1"
})(["width:100%;height:0.25rem;background-color:", ";border-radius:0.125rem;"], _plasmaCore.surfaceLiquid02);

var StyledProgress = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web__sc-1e491uq-2"
})(["width:", "%;height:100%;", " border-radius:0.125rem;"], function (_ref) {
  var value = _ref.value;
  return value;
}, function (_ref2) {
  var status = _ref2.status;
  return status && statuses[status];
});

var StyledValue = /*#__PURE__*/_styledComponents["default"].span.withConfig({
  componentId: "plasma-web__sc-1e491uq-3"
})(["", ";margin-left:0.5rem;color:", ";"], _plasmaCore.caption, _plasmaCore.tertiary);
/**
 * Компонент для отображения прогресса в процентах.
 */


var Progress = function Progress(_ref3) {
  var value = _ref3.value,
      status = _ref3.status,
      _ref3$displayValue = _ref3.displayValue,
      displayValue = _ref3$displayValue === void 0 ? true : _ref3$displayValue;
  var min = 0;
  var max = 100;
  var normalizedValue = Math.max(Math.min(value, max), min);
  return /*#__PURE__*/_react["default"].createElement(StyledRoot, null, /*#__PURE__*/_react["default"].createElement(StyledTrack, null, /*#__PURE__*/_react["default"].createElement(StyledProgress, {
    value: normalizedValue,
    status: status
  })), displayValue && /*#__PURE__*/_react["default"].createElement(StyledValue, null, normalizedValue, "%"));
};

exports.Progress = Progress;
