// @sberdevices/plasma-tokens-web/es/themesValues/dark.js
// Generated by robots, do not change this manually!
/** Базовый белый цвет совпадает с whitePrimary */
export var white = '#FFFFFF';
/** Первичный белый цвет */
export var whitePrimary = '#FFFFFF';
/** Вторичный белый цвет */
export var whiteSecondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный белый цвет */
export var whiteTertiary = 'rgba(255, 255, 255, 0.28)';
/** Базовый черный цвет совпадает с blackPrimary */
export var black = '#080808';
/** Первичный черный цвет */
export var blackPrimary = '#080808';
/** Вторичный черныйцвет */
export var blackSecondary = 'rgba(8, 8, 8, 0.56)';
/** Третичный черный цвет */
export var blackTertiary = 'rgba(8, 8, 8, 0.28)';
/** Чёрный непрозрачный, не зависит от темы */
export var dark01 = '#171717';
/** Чёрный непрозрачный, не зависит от темы */
export var dark02 = '#232323';
/** Чёрный непрозрачный, не зависит от темы */
export var dark03 = '#363636';
/** Прозрачный цвет */
export var transparent = 'rgba(0, 0, 0, 0)';
/** Второстепенная прозрачная */
export var buttonClear = 'rgba(0, 0, 0, 0)';
/** Дефолтная чёрная, не зависит от темы */
export var buttonBlack = '#080808';
/** Второстепенная чёрная, не зависит от темы */
export var buttonBlackSecondary = 'rgba(8, 8, 8, 0.12)';
/** Чёрная для использования поверх картинок */
export var buttonBlackTransparent = 'rgba(8, 8, 8, 0.56)';
/** Дефолтная белая, не зависит от темы */
export var buttonWhite = '#FFFFFF';
/** Второстепенная белая, не зависит от темы */
export var buttonWhiteSecondary = 'rgba(255, 255, 255, 0.12)';
/** Базовый цвет текста, совпадает с primary */
export var text = 'rgba(255, 255, 255, 0.96)';
export var link = '#3F81FD';
export var linkHover = '#6298FD';
export var linkActive = '#1767FD';
export var linkVisited = '#5993FF';
export var linkVisitedHover = '#7DAAFF';
export var linkVisitedActive = '#3078FF';
/** Основной текст в интерфейсе */
export var primary = 'rgba(255, 255, 255, 0.96)';
/** Второстепенный текст */
export var secondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный цвет текста */
export var tertiary = 'rgba(255, 255, 255, 0.28)';
export var paragraph = 'rgba(255, 255, 255, 0.96)';
export var inverse = '#080808';
/** Основной бэкграунд всех приложений и смартапов */
export var background = '#080808';
/** Первичный цвет фона */
export var backgroundPrimary = '#171717';
/** Вторичный цвет фона */
export var backgroundSecondary = '#232323';
/** Третичный цвет фона */
export var backgroundTertiary = '#363636';
export var inputBorder = 'rgba(255, 255, 255, 0.16)';
export var inputBorderHover = 'rgba(255, 255, 255, 0.32)';
export var inputBorderFocus = '#5993FF';
/** Акцентный цвет бренда */
export var accent = '#5993FF';
/** Обозначение успешного сценария */
export var success = '#24B23E';
/** Цвет предупреждения */
export var warning = '#FA6D20';
/** Цвет ошибки */
export var critical = '#FF4D5F';
/** Цвет фона паранжи */
export var overlay = 'rgba(0, 0, 0, 0.8)';
/** Полупрозрачные поверхности */
export var surfaceLiquid01 = 'rgba(255, 255, 255, 0.06)';
/** Полупрозрачные поверхности */
export var surfaceLiquid02 = 'rgba(255, 255, 255, 0.12)';
/** Полупрозрачные поверхности */
export var surfaceLiquid03 = 'rgba(255, 255, 255, 0.2)';
/** Непрозрачные поверхности */
export var surfaceSolid01 = '#171717';
/** Непрозрачные поверхности */
export var surfaceSolid02 = '#232323';
/** Непрозрачные поверхности */
export var surfaceSolid03 = '#363636';
/** Основной фон для карточек */
export var surfaceCard = 'rgba(255, 255, 255, 0.12)';
/** Первичный цвет контролов */
export var buttonPrimary = '#3F82FD';
export var buttonPrimaryHover = '#538efe';
export var buttonPrimaryActive = '#3a75e5';
/** Второстепенная кнопка */
export var buttonSecondary = 'rgba(255, 255, 255, 0.12)';
export var buttonSecondaryHover = 'rgba(255, 255, 255, 0.16)';
export var buttonSecondaryActive = 'rgba(255, 255, 255, 0.1)';
/** Акцентный цвет у контролов */
export var buttonAccent = '#3F82FD';
/** Кнопка для успешного сценария */
export var buttonSuccess = '#21A038';
export var buttonSuccessHover = '#3eaa4d';
export var buttonSuccessActive = '#1d9233';
/** Цвет предупреждения у контролов */
export var buttonWarning = '#E35502';
export var buttonWarningHover = '#f76d2f';
export var buttonWarningActive = '#de550a';
/** Цвет ошибки у контролов */
export var buttonCritical = '#FF2E43';
export var buttonCriticalHover = '#ff4757';
export var buttonCriticalActive = '#e82a3e';
/** Цвет зажатого контрола */
export var buttonChecked = '#FFFFFF';
export var buttonCheckedHover = '#FFFFFF';
export var buttonCheckedHoverColor = 'hsla(0, 0%, 3.1372549019607843%, 1)';
export var buttonCheckedActive = '#FFFFFF';
export var buttonCheckedActiveColor = 'hsla(0, 0%, 3.1372549019607843%, 1)';
/** Цвет рамки фокуса у контрола */
export var buttonFocused = '#3F82FD';
/** Градиент для заливки основного фона */
export var gradient = 'linear-gradient(45deg, hsla(219.87577639751552, 100%, 68.43137254901961%, 1) 0%, hsla(190.1639344262295, 94.81865284974093%, 62.15686274509804%, 1) 100%)';
export var gradientDevice = '';
export var voicePhraseGradient = '';
export var skeletonGradient = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% )';
export var skeletonGradientLighter = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% )';
export var speechBubbleSent = '';
export var speechBubbleReceived = '';
