// @sberdevices/plasma-ui/es/components/TextField/TextField.js
var _excluded = ["id", "size", "disabled", "status", "label", "placeholder", "contentLeft", "contentRight", "helperText", "style", "className"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { forwardRef } from 'react';
import styled from 'styled-components';
import { FieldRoot, FieldPlaceholder, FieldContent, Input } from '@sberdevices/plasma-core';
import { FieldHelper, applyInputStyles } from '../Field';
var StyledInput = /*#__PURE__*/styled(Input).withConfig({
  componentId: "plasma-ui__sc-1vvf73-0"
})(["", " border-radius:1rem;"], applyInputStyles);
/**
 * Поле ввода текста.
 */
// eslint-disable-next-line prefer-arrow-callback

export var TextField = /*#__PURE__*/forwardRef(function TextField(_ref, ref) {
  var id = _ref.id,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 'l' : _ref$size,
      disabled = _ref.disabled,
      status = _ref.status,
      label = _ref.label,
      placeholder = _ref.placeholder,
      contentLeft = _ref.contentLeft,
      contentRight = _ref.contentRight,
      helperText = _ref.helperText,
      style = _ref.style,
      className = _ref.className,
      rest = _objectWithoutProperties(_ref, _excluded);

  var placeLabel = label || placeholder;
  return /*#__PURE__*/React.createElement(FieldRoot, {
    $size: size,
    $disabled: disabled,
    $isContentLeft: Boolean(contentLeft),
    $isContentRight: Boolean(contentRight),
    $isHelper: Boolean(helperText),
    status: status,
    style: style,
    className: className
  }, contentLeft && /*#__PURE__*/React.createElement(FieldContent, {
    pos: "left"
  }, contentLeft), /*#__PURE__*/React.createElement(StyledInput, _extends({
    $size: size,
    ref: ref,
    id: id,
    placeholder: placeLabel,
    disabled: disabled,
    status: status,
    "aria-describedby": id ? "".concat(id, "-helpertext") : undefined
  }, rest)), placeLabel && size === 'l' && /*#__PURE__*/React.createElement(FieldPlaceholder, {
    htmlFor: id
  }, placeLabel), contentRight && /*#__PURE__*/React.createElement(FieldContent, {
    pos: "right"
  }, contentRight), helperText && /*#__PURE__*/React.createElement(FieldHelper, {
    id: id ? "".concat(id, "-helpertext") : undefined
  }, helperText));
});
