// @sberdevices/plasma-core/components/Carousel/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "CarouselGridWrapper", {
  enumerable: true,
  get: function get() {
    return _Carousel.CarouselGridWrapper;
  }
});
Object.defineProperty(exports, "Carousel", {
  enumerable: true,
  get: function get() {
    return _Carousel.Carousel;
  }
});
Object.defineProperty(exports, "CarouselTrack", {
  enumerable: true,
  get: function get() {
    return _Carousel.CarouselTrack;
  }
});
Object.defineProperty(exports, "CarouselContext", {
  enumerable: true,
  get: function get() {
    return _CarouselContext.CarouselContext;
  }
});
Object.defineProperty(exports, "CarouselItem", {
  enumerable: true,
  get: function get() {
    return _CarouselItem.CarouselItem;
  }
});
Object.defineProperty(exports, "useCarousel", {
  enumerable: true,
  get: function get() {
    return _hooks.useCarousel;
  }
});
Object.defineProperty(exports, "useCarouselItem", {
  enumerable: true,
  get: function get() {
    return _hooks.useCarouselItem;
  }
});
Object.defineProperty(exports, "useCarouselContext", {
  enumerable: true,
  get: function get() {
    return _hooks.useCarouselContext;
  }
});

var _Carousel = /*#__PURE__*/require("./Carousel");

var _CarouselContext = /*#__PURE__*/require("./CarouselContext");

var _CarouselItem = /*#__PURE__*/require("./CarouselItem");

var _hooks = /*#__PURE__*/require("./hooks");
