// @salutejs/plasma-ui/components/Button/Button.perftest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ActionButton = exports.Default = void 0;

var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));

var _plasmaIcons = /*#__PURE__*/require("@salutejs/plasma-icons");

var _ = /*#__PURE__*/require(".");

var _IconMic, _IconMic2;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var defaultProps = {
  size: 'm',
  view: 'primary',
  pin: 'square-square',
  enableIcon: true,
  scaleOnInteraction: true,
  outlined: true,
  focused: false,
  disabled: false,
  square: false,
  stretch: false,
  isLoading: false,
  onClick: function onClick() {},
  onFocus: function onFocus() {},
  onBlur: function onBlur() {}
};
var text = 'Hello Plasma';

var Default = function Default() {
  return /*#__PURE__*/_react["default"].createElement(_.Button, _extends({
    text: text,
    contentLeft: _IconMic || (_IconMic = /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconMic, {
      size: "s",
      color: "inherit"
    }))
  }, defaultProps));
};

exports.Default = Default;
var actionButton = {
  size: 'm',
  view: 'primary',
  pin: 'square-square',
  scaleOnInteraction: true,
  disabled: false,
  tabIndex: 0,
  onClick: function onClick() {},
  onFocus: function onFocus() {},
  onBlur: function onBlur() {}
};

var ActionButton = function ActionButton() {
  return /*#__PURE__*/_react["default"].createElement(_.ActionButton, actionButton, _IconMic2 || (_IconMic2 = /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconMic, {
    size: "xs",
    color: "inherit"
  })));
};

exports.ActionButton = ActionButton;
