// @sberdevices/plasma-typo/lib/cjs/standard.js
"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.standard = void 0;
var styled_components_1 = require("styled-components");
var helpers_1 = require("./helpers");
var typoS = helpers_1.prepareStandardBreakpointTypo({
    'dspl-l': {
        'font-size': '5.5rem',
        'font-weight': '600',
        'line-height': '5.75rem',
    },
    'dspl-m': {
        'font-size': '3.5rem',
        'font-weight': '600',
        'line-height': '3.875rem',
    },
    'dspl-s': {
        'font-size': '2.5rem',
        'font-weight': '600',
        'line-height': '2.875rem',
    },
    h1: {
        'font-size': '1.75rem',
        'font-weight': '600',
        'line-height': '2.125rem',
    },
    h2: {
        'font-size': '1.5rem',
        'font-weight': '600',
        'line-height': '1.875rem',
    },
    h3: {
        'font-size': '1.25rem',
        'font-weight': '600',
        'line-height': '1.625rem',
    },
    h4: {
        'font-size': '1.125rem',
        'font-weight': '600',
        'line-height': '1.5rem',
    },
    h5: {
        'font-size': '0.875rem',
        'font-weight': '600',
        'line-height': '1.375rem',
    },
    'body-l': {
        'font-size': '1.125rem',
        'font-weight': '400',
        'line-height': '1.375rem',
    },
    'body-l-bold': {
        'font-size': '1.125rem',
        'font-weight': '600',
        'line-height': '1.375rem',
    },
    'body-m': {
        'font-size': '1rem',
        'font-weight': '400',
        'line-height': '1.25rem',
    },
    'body-m-bold': {
        'font-size': '1rem',
        'font-weight': '600',
        'line-height': '1.25rem',
    },
    'body-s': {
        'font-size': '0.875rem',
        'font-weight': '400',
        'line-height': '1.125rem',
    },
    'body-s-bold': {
        'font-size': '0.875rem',
        'font-weight': '600',
        'line-height': '1.125rem',
    },
    'body-xs': {
        'font-size': '0.75rem',
        'font-weight': '400',
        'line-height': '0.875rem',
    },
    'body-xs-bold': {
        'font-size': '0.75rem',
        'font-weight': '600',
        'line-height': '0.875rem',
    },
    'body-xxs': {
        'font-size': '0.625rem',
        'font-weight': '400',
        'line-height': '0.75rem',
    },
    'body-xxs-bold': {
        'font-size': '0.625rem',
        'font-weight': '600',
        'line-height': '0.75rem',
    },
    'text-l': {
        'font-size': '1.125rem',
        'font-weight': '400',
        'line-height': '1.625rem',
    },
    'text-l-bold': {
        'font-size': '1.125rem',
        'font-weight': '600',
        'line-height': '1.625rem',
    },
    'text-m': {
        'font-size': '1rem',
        'font-weight': '400',
        'line-height': '1.5rem',
    },
    'text-m-bold': {
        'font-size': '1rem',
        'font-weight': '600',
        'line-height': '1.5rem',
    },
    'text-s': {
        'font-size': '0.875rem',
        'font-weight': '400',
        'line-height': '1.25rem',
    },
    'text-s-bold': {
        'font-size': '0.875rem',
        'font-weight': '600',
        'line-height': '1.25rem',
    },
    'text-xs': {
        'font-size': '0.75rem',
        'font-weight': '400',
        'line-height': '1rem',
    },
    'text-xs-bold': {
        'font-size': '0.75rem',
        'font-weight': '600',
        'line-height': '1rem',
    },
});
var typoM = helpers_1.prepareStandardBreakpointTypo({
    'dspl-l': {
        'font-size': '7rem',
        'font-weight': '600',
        'line-height': '7rem',
    },
    'dspl-m': {
        'font-size': '4.5rem',
        'font-weight': '600',
        'line-height': '4.75rem',
    },
    'dspl-s': {
        'font-size': '3rem',
        'font-weight': '600',
        'line-height': '3.375rem',
    },
    h1: {
        'font-size': '2.5rem',
        'font-weight': '600',
        'line-height': '2.875rem',
    },
    h2: {
        'font-size': '1.75rem',
        'font-weight': '600',
        'line-height': '2.125rem',
    },
    h3: {
        'font-size': '1.25rem',
        'font-weight': '600',
        'line-height': '1.625rem',
    },
    h4: {
        'font-size': '1.125rem',
        'font-weight': '600',
        'line-height': '1.5rem',
    },
    h5: {
        'font-size': '1rem',
        'font-weight': '600',
        'line-height': '1.375rem',
    },
    'body-l': {
        'font-size': '1.125rem',
        'font-weight': '400',
        'line-height': '1.375rem',
    },
    'body-l-bold': {
        'font-size': '1.125rem',
        'font-weight': '600',
        'line-height': '1.375rem',
    },
    'body-m': {
        'font-size': '1rem',
        'font-weight': '400',
        'line-height': '1.25rem',
    },
    'body-m-bold': {
        'font-size': '1rem',
        'font-weight': '600',
        'line-height': '1.25rem',
    },
    'body-s': {
        'font-size': '0.875rem',
        'font-weight': '400',
        'line-height': '1.125rem',
    },
    'body-s-bold': {
        'font-size': '0.875rem',
        'font-weight': '600',
        'line-height': '1.125rem',
    },
    'body-xs': {
        'font-size': '0.75rem',
        'font-weight': '400',
        'line-height': '0.875rem',
    },
    'body-xs-bold': {
        'font-size': '0.75rem',
        'font-weight': '600',
        'line-height': '0.875rem',
    },
    'body-xxs': {
        'font-size': '0.625rem',
        'font-weight': '400',
        'line-height': '0.75rem',
    },
    'body-xxs-bold': {
        'font-size': '0.625rem',
        'font-weight': '600',
        'line-height': '0.75rem',
    },
    'text-l': {
        'font-size': '1.25rem',
        'font-weight': '400',
        'line-height': '1.75rem',
    },
    'text-l-bold': {
        'font-size': '1.25rem',
        'font-weight': '600',
        'line-height': '1.75rem',
    },
    'text-m': {
        'font-size': '1rem',
        'font-weight': '400',
        'line-height': '1.5rem',
    },
    'text-m-bold': {
        'font-size': '1rem',
        'font-weight': '600',
        'line-height': '1.5rem',
    },
    'text-s': {
        'font-size': '0.875rem',
        'font-weight': '400',
        'line-height': '1.25rem',
    },
    'text-s-bold': {
        'font-size': '0.875rem',
        'font-weight': '600',
        'line-height': '1.25rem',
    },
    'text-xs': {
        'font-size': '0.75rem',
        'font-weight': '400',
        'line-height': '1rem',
    },
    'text-xs-bold': {
        'font-size': '0.75rem',
        'font-weight': '600',
        'line-height': '1rem',
    },
});
var typoL = helpers_1.prepareStandardBreakpointTypo({
    'dspl-l': {
        'font-size': '8rem',
        'font-weight': '600',
        'line-height': '8rem',
    },
    'dspl-m': {
        'font-size': '5.5rem',
        'font-weight': '600',
        'line-height': '5.75rem',
    },
    'dspl-s': {
        'font-size': '4rem',
        'font-weight': '600',
        'line-height': '4.25rem',
    },
    h1: {
        'font-size': '3rem',
        'font-weight': '600',
        'line-height': '3.375rem',
    },
    h2: {
        'font-size': '2rem',
        'font-weight': '600',
        'line-height': '2.375rem',
    },
    h3: {
        'font-size': '1.5rem',
        'font-weight': '600',
        'line-height': '1.875rem',
    },
    h4: {
        'font-size': '1.25rem',
        'font-weight': '600',
        'line-height': '1.625rem',
    },
    h5: {
        'font-size': '1.125rem',
        'font-weight': '600',
        'line-height': '1.5rem',
    },
    'body-l': {
        'font-size': '1.125rem',
        'font-weight': '400',
        'line-height': '1.375rem',
    },
    'body-l-bold': {
        'font-size': '1.125rem',
        'font-weight': '600',
        'line-height': '1.375rem',
    },
    'body-m': {
        'font-size': '1rem',
        'font-weight': '400',
        'line-height': '1.25rem',
    },
    'body-m-bold': {
        'font-size': '1rem',
        'font-weight': '600',
        'line-height': '1.25rem',
    },
    'body-s': {
        'font-size': '0.875rem',
        'font-weight': '400',
        'line-height': '1.125rem',
    },
    'body-s-bold': {
        'font-size': '0.875rem',
        'font-weight': '600',
        'line-height': '1.125rem',
    },
    'body-xs': {
        'font-size': '0.75rem',
        'font-weight': '400',
        'line-height': '0.875rem',
    },
    'body-xs-bold': {
        'font-size': '0.75rem',
        'font-weight': '600',
        'line-height': '0.875rem',
    },
    'body-xxs': {
        'font-size': '0.625rem',
        'font-weight': '400',
        'line-height': '0.75rem',
    },
    'body-xxs-bold': {
        'font-size': '0.625rem',
        'font-weight': '600',
        'line-height': '0.75rem',
    },
    'text-l': {
        'font-size': '1.5rem',
        'font-weight': '400',
        'line-height': '2rem',
    },
    'text-l-bold': {
        'font-size': '1.5rem',
        'font-weight': '600',
        'line-height': '2rem',
    },
    'text-m': {
        'font-size': '1.125rem',
        'font-weight': '400',
        'line-height': '1.625rem',
    },
    'text-m-bold': {
        'font-size': '1.125rem',
        'font-weight': '600',
        'line-height': '1.625rem',
    },
    'text-s': {
        'font-size': '0.875rem',
        'font-weight': '400',
        'line-height': '1.25rem',
    },
    'text-s-bold': {
        'font-size': '0.875rem',
        'font-weight': '600',
        'line-height': '1.25rem',
    },
    'text-xs': {
        'font-size': '0.75rem',
        'font-weight': '400',
        'line-height': '1rem',
    },
    'text-xs-bold': {
        'font-size': '0.75rem',
        'font-weight': '600',
        'line-height': '1rem',
    },
});
exports.standard = function (_a) {
    var _b = _a.deviceScale, deviceScale = _b === void 0 ? 1 : _b;
    return styled_components_1.css(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    :root {\n        --plasma-typo-display-font-family: 'SB Sans Display', sans-serif;\n        --plasma-typo-text-font-family: 'SB Sans Text', sans-serif;\n        --plasma-typo-dspl-l-font-family: var(--plasma-typo-display-font-family);\n        --plasma-typo-dspl-l-font-style: normal;\n        --plasma-typo-dspl-l-letter-spacing: normal;\n        --plasma-typo-dspl-m-font-family: var(--plasma-typo-display-font-family);\n        --plasma-typo-dspl-m-font-style: normal;\n        --plasma-typo-dspl-m-letter-spacing: normal;\n        --plasma-typo-dspl-s-font-family: var(--plasma-typo-display-font-family);\n        --plasma-typo-dspl-s-font-style: normal;\n        --plasma-typo-dspl-s-letter-spacing: normal;\n        --plasma-typo-h1-font-family: var(--plasma-typo-display-font-family);\n        --plasma-typo-h1-font-style: normal;\n        --plasma-typo-h1-letter-spacing: normal;\n        --plasma-typo-h2-font-family: var(--plasma-typo-display-font-family);\n        --plasma-typo-h2-font-style: normal;\n        --plasma-typo-h2-letter-spacing: normal;\n        --plasma-typo-h3-font-family: var(--plasma-typo-display-font-family);\n        --plasma-typo-h3-font-style: normal;\n        --plasma-typo-h3-letter-spacing: normal;\n        --plasma-typo-h4-font-family: var(--plasma-typo-display-font-family);\n        --plasma-typo-h4-font-style: normal;\n        --plasma-typo-h4-letter-spacing: normal;\n        --plasma-typo-h5-font-family: var(--plasma-typo-display-font-family);\n        --plasma-typo-h5-font-style: normal;\n        --plasma-typo-h5-letter-spacing: normal;\n        --plasma-typo-body-l-font-family: var(--plasma-typo-text-font-family);\n        --plasma-typo-body-l-font-style: normal;\n        --plasma-typo-body-l-letter-spacing: normal;\n        --plasma-typo-body-m-font-family: var(--plasma-typo-text-font-family);\n        --plasma-typo-body-m-font-style: normal;\n        --plasma-typo-body-m-letter-spacing: normal;\n        --plasma-typo-body-s-font-family: var(--plasma-typo-text-font-family);\n        --plasma-typo-body-s-font-style: normal;\n        --plasma-typo-body-s-letter-spacing: normal;\n        --plasma-typo-body-xs-font-family: var(--plasma-typo-text-font-family);\n        --plasma-typo-body-xs-font-style: normal;\n        --plasma-typo-body-xs-letter-spacing: normal;\n        --plasma-typo-body-xxs-font-family: var(--plasma-typo-text-font-family);\n        --plasma-typo-body-xxs-font-style: normal;\n        --plasma-typo-body-xxs-letter-spacing: normal;\n        --plasma-typo-text-l-font-family: var(--plasma-typo-text-font-family);\n        --plasma-typo-text-l-font-style: normal;\n        --plasma-typo-text-l-letter-spacing: -0.019em;\n        --plasma-typo-text-m-font-family: var(--plasma-typo-text-font-family);\n        --plasma-typo-text-m-font-style: normal;\n        --plasma-typo-text-m-letter-spacing: -0.019em;\n        --plasma-typo-text-s-font-family: var(--plasma-typo-text-font-family);\n        --plasma-typo-text-s-font-style: normal;\n        --plasma-typo-text-s-letter-spacing: -0.019em;\n        --plasma-typo-text-xs-font-family: var(--plasma-typo-text-font-family);\n        --plasma-typo-text-xs-font-style: normal;\n        --plasma-typo-text-xs-letter-spacing: -0.019em;\n\n        font-size: ", "px;\n\n        @media (max-width: ", "px) {\n            ", "\n        }\n\n        @media (min-width: ", "px) and (max-width: ", "px) {\n            ", "\n        }\n\n        @media (min-width: ", "px) {\n            ", "\n        }\n    }\n"], ["\n    :root {\n        --plasma-typo-display-font-family: 'SB Sans Display', sans-serif;\n        --plasma-typo-text-font-family: 'SB Sans Text', sans-serif;\n        --plasma-typo-dspl-l-font-family: var(--plasma-typo-display-font-family);\n        --plasma-typo-dspl-l-font-style: normal;\n        --plasma-typo-dspl-l-letter-spacing: normal;\n        --plasma-typo-dspl-m-font-family: var(--plasma-typo-display-font-family);\n        --plasma-typo-dspl-m-font-style: normal;\n        --plasma-typo-dspl-m-letter-spacing: normal;\n        --plasma-typo-dspl-s-font-family: var(--plasma-typo-display-font-family);\n        --plasma-typo-dspl-s-font-style: normal;\n        --plasma-typo-dspl-s-letter-spacing: normal;\n        --plasma-typo-h1-font-family: var(--plasma-typo-display-font-family);\n        --plasma-typo-h1-font-style: normal;\n        --plasma-typo-h1-letter-spacing: normal;\n        --plasma-typo-h2-font-family: var(--plasma-typo-display-font-family);\n        --plasma-typo-h2-font-style: normal;\n        --plasma-typo-h2-letter-spacing: normal;\n        --plasma-typo-h3-font-family: var(--plasma-typo-display-font-family);\n        --plasma-typo-h3-font-style: normal;\n        --plasma-typo-h3-letter-spacing: normal;\n        --plasma-typo-h4-font-family: var(--plasma-typo-display-font-family);\n        --plasma-typo-h4-font-style: normal;\n        --plasma-typo-h4-letter-spacing: normal;\n        --plasma-typo-h5-font-family: var(--plasma-typo-display-font-family);\n        --plasma-typo-h5-font-style: normal;\n        --plasma-typo-h5-letter-spacing: normal;\n        --plasma-typo-body-l-font-family: var(--plasma-typo-text-font-family);\n        --plasma-typo-body-l-font-style: normal;\n        --plasma-typo-body-l-letter-spacing: normal;\n        --plasma-typo-body-m-font-family: var(--plasma-typo-text-font-family);\n        --plasma-typo-body-m-font-style: normal;\n        --plasma-typo-body-m-letter-spacing: normal;\n        --plasma-typo-body-s-font-family: var(--plasma-typo-text-font-family);\n        --plasma-typo-body-s-font-style: normal;\n        --plasma-typo-body-s-letter-spacing: normal;\n        --plasma-typo-body-xs-font-family: var(--plasma-typo-text-font-family);\n        --plasma-typo-body-xs-font-style: normal;\n        --plasma-typo-body-xs-letter-spacing: normal;\n        --plasma-typo-body-xxs-font-family: var(--plasma-typo-text-font-family);\n        --plasma-typo-body-xxs-font-style: normal;\n        --plasma-typo-body-xxs-letter-spacing: normal;\n        --plasma-typo-text-l-font-family: var(--plasma-typo-text-font-family);\n        --plasma-typo-text-l-font-style: normal;\n        --plasma-typo-text-l-letter-spacing: -0.019em;\n        --plasma-typo-text-m-font-family: var(--plasma-typo-text-font-family);\n        --plasma-typo-text-m-font-style: normal;\n        --plasma-typo-text-m-letter-spacing: -0.019em;\n        --plasma-typo-text-s-font-family: var(--plasma-typo-text-font-family);\n        --plasma-typo-text-s-font-style: normal;\n        --plasma-typo-text-s-letter-spacing: -0.019em;\n        --plasma-typo-text-xs-font-family: var(--plasma-typo-text-font-family);\n        --plasma-typo-text-xs-font-style: normal;\n        --plasma-typo-text-xs-letter-spacing: -0.019em;\n\n        font-size: ", "px;\n\n        @media (max-width: ", "px) {\n            ", "\n        }\n\n        @media (min-width: ", "px) and (max-width: ", "px) {\n            ", "\n        }\n\n        @media (min-width: ", "px) {\n            ", "\n        }\n    }\n"])), 16 * deviceScale, 559 * deviceScale, typoS, 560 * deviceScale, 959 * deviceScale, typoM, 960 * deviceScale, typoL);
};
var templateObject_1;
