// @salutejs/plasma-tokens/brands/stylesSalute/index.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.buttonCritical = exports.buttonWarning = exports.buttonSuccess = exports.buttonSecondary = exports.surfaceCard = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.surfaceSolid01 = exports.surfaceLiquid03 = exports.surfaceLiquid02 = exports.surfaceLiquid01 = exports.critical = exports.warning = exports.success = exports.background = exports.paragraph = exports.tertiary = exports.secondary = exports.primary = exports.text = exports.buttonWhiteSecondary = exports.buttonWhite = exports.buttonBlackSecondary = exports.buttonBlack = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.skeletonGradientLighter = exports.skeletonGradient = exports.backgroundMobileAssistantMain = exports.backgroundMobileAssistantBrand = exports.backgroundMobileAssistantB2E = exports.backgroundMobileAssistantJoy = exports.backgroundMobileAssistantAthena = exports.backgroundModalSheet = exports.backgroundSecondary = exports.backgroundPrimary = exports.onLightSurfaceGradientMain = exports.onLightSurfaceAccentMain = exports.onLightSurfaceTransparentDeep = exports.onLightSurfaceAccentBrand = exports.onLightSurfaceAccentB2E = exports.onLightSurfaceAccentJoy = exports.onLightSurfaceAccentAthena = exports.onLightSurfaceNegative = exports.onLightSurfaceWarning = exports.onLightSurfacePositive = exports.onLightSurfaceTransparentCard = exports.onLightSurfaceTransparentTertiary = exports.onLightSurfaceTransparentSecondary = exports.onLightSurfaceTransparentPrimary = exports.onLightSurfaceSolidCard = exports.onLightSurfaceSolidTertiary = exports.onLightSurfaceSolidSecondary = exports.onLightSurfaceSolidPrimary = exports.onLightSurfaceSolidDefault = exports.onDarkSurfaceGradientMain = exports.onDarkSurfaceAccentMain = exports.onDarkSurfaceAccentBrand = exports.onDarkSurfaceAccentB2E = exports.onDarkSurfaceAccentJoy = exports.onDarkSurfaceAccentAthena = exports.onDarkSurfaceNegative = exports.onDarkSurfaceWarning = exports.onDarkSurfacePositive = exports.onDarkSurfaceTransparentCard = exports.onDarkSurfaceTransparentTertiary = exports.onDarkSurfaceTransparentSecondary = exports.onDarkSurfaceTransparentPrimary = exports.onDarkSurfaceSolidCard = exports.onDarkSurfaceSolidTertiary = exports.onDarkSurfaceSolidSecondary = exports.onDarkSurfaceSolidPrimary = exports.onDarkSurfaceSolidDefault = exports.surfaceGradientMain = exports.surfaceAccentMain = exports.surfaceSkeleton = exports.surfaceSpeechBubbleReceived = exports.surfaceSpeechBubbleSent = exports.surfaceAccentBrand = exports.surfaceAccentB2E = exports.surfaceAccentJoy = exports.surfaceAccentAthena = exports.surfaceClear = exports.surfaceNegative = exports.surfaceWarning = exports.surfacePositive = exports.surfaceTransparentCard = exports.surfaceTransparentTertiary = exports.surfaceTransparentSecondary = exports.surfaceTransparentPrimary = exports.surfaceSolidCard = exports.surfaceSolidTertiary = exports.surfaceSolidSecondary = exports.surfaceSolidPrimary = exports.surfaceSolidDefault = exports.inverseTextParagraph = exports.inverseTextTertiary = exports.inverseTextSecondary = exports.inverseTextPrimary = exports.onLightTextAccentMain = exports.onLightTextGradientMain = exports.onLightTextGradientBrand = exports.onLightTextGradientB2E = exports.onLightTextGradientJoy = exports.onLightTextGradientAthena = exports.onLightTextAccentBrand = exports.onLightTextAccentB2E = exports.onLightTextAccentJoy = exports.onLightTextAccentAthena = exports.onLightTextNegative = exports.onLightTextWarning = exports.onLightTextPositive = exports.onLightTextParagraph = exports.onLightTextTertiary = exports.onLightTextSecondary = exports.onLightTextPrimary = exports.onDarkTextAccentMain = exports.onDarkTextGradientMain = exports.onDarkTextGradientBrand = exports.onDarkTextGradientB2E = exports.onDarkTextGradientJoy = exports.onDarkTextGradientAthena = exports.onDarkTextAccentBrand = exports.onDarkTextAccentB2E = exports.onDarkTextAccentJoy = exports.onDarkTextAccentAthena = exports.onDarkTextNegative = exports.onDarkTextWarning = exports.onDarkTextPositive = exports.onDarkTextParagraph = exports.onDarkTextTertiary = exports.onDarkTextSecondary = exports.onDarkTextPrimary = exports.textAccentMain = exports.textGradientMain = exports.textGradientBrand = exports.textGradientB2E = exports.textGradientJoy = exports.textGradientAthena = exports.textAccentBrand = exports.textAccentB2E = exports.textAccentJoy = exports.textAccentAthena = exports.textNegative = exports.textWarning = exports.textPositive = exports.textParagraph = exports.textTertiary = exports.textSecondary = exports.textPrimary = exports.onDarkOverlayBlur = exports.onDarkOverlayHard = exports.onDarkOverlaySoft = void 0;
/** Цвет фона паранжи светлый на темном фоне */
exports.onDarkOverlaySoft = 'var(--on-dark-overlay-soft, rgba(0, 0, 0, 0.56))';
/** Цвет фона паранжи темный на темном фоне */
exports.onDarkOverlayHard = 'var(--on-dark-overlay-hard, #000000EB)';
/** Оверлей с блюром */
exports.onDarkOverlayBlur = 'var(--on-dark-overlay-blur, #26262647)';
/** Основной цвет текста */
exports.textPrimary = 'var(--text-primary, #FFFFFFF5)';
/** Вторичный цвет текста */
exports.textSecondary = 'var(--text-secondary, #FFFFFF8F)';
/** Третичный цвет текста */
exports.textTertiary = 'var(--text-tertiary, #FFFFFF47)';
/** Сплошной наборный текст */
exports.textParagraph = 'var(--text-paragraph, #FFFFFFCC)';
/** Цвет успеха */
exports.textPositive = 'var(--text-positive, #24B23E)';
/** Цвет предупреждения */
exports.textWarning = 'var(--text-warning, #FF7024)';
/** Цвет ошибки */
exports.textNegative = 'var(--text-negative, #FF293E)';
/** Акцентный цвет Афины */
exports.textAccentAthena = 'var(--text-accent-athena, #199AF0)';
/** Акцентный цвет Джой */
exports.textAccentJoy = 'var(--text-accent-joy, #C46BFF)';
/** Акцентный цвет B2E-ассистента */
exports.textAccentB2E = 'var(--text-accent-b2-e, #24B23E)';
/** Акцентный цвет бренда */
exports.textAccentBrand = 'var(--text-accent-brand, #528EFF)';
/** Градиент Афины */
exports.textGradientAthena = 'var(--text-gradient-athena, linear-gradient(93.97deg, #52BAFF 6.49%, #ADADFF 93.51%))';
/** Градиент Джой */
exports.textGradientJoy = 'var(--text-gradient-joy, linear-gradient(93.97deg, #FD9C68 6.49%, #F391E2 93.51%))';
/** Градиент B2E-ассистента */
exports.textGradientB2E = 'var(--text-gradient-b2-e, linear-gradient(89.83deg, #24B23E 0%, #1CC9C9 100%))';
/** Градиент бренда */
exports.textGradientBrand = 'var(--text-gradient-brand, linear-gradient(89.83deg, #528EFF 0%, #04C6C9 100%))';
/** Градиент */
exports.textGradientMain = 'var(--text-gradient-main, linear-gradient(89.83deg, #24B23E 0%, #1CC9C9 100%))';
/** Акцентный цвет */
exports.textAccentMain = 'var(--text-accent-main, #24B23E)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimary = 'var(--on-dark-text-primary, #FFFFFFF5)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondary = 'var(--on-dark-text-secondary, #FFFFFF8F)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiary = 'var(--on-dark-text-tertiary, #FFFFFF47)';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraph = 'var(--on-dark-text-paragraph, #FFFFFFCC)';
/** Цвет успеха на темном фоне */
exports.onDarkTextPositive = 'var(--on-dark-text-positive, #24B23E)';
/** Цвет предупреждения на темном фоне */
exports.onDarkTextWarning = 'var(--on-dark-text-warning, #FF7024)';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegative = 'var(--on-dark-text-negative, #FF293E)';
/** Акцентный цвет Афины на темном */
exports.onDarkTextAccentAthena = 'var(--on-dark-text-accent-athena, #199AF0)';
/** Акцентный цвет Джой на темном фоне */
exports.onDarkTextAccentJoy = 'var(--on-dark-text-accent-joy, #C46BFF)';
/** Акцентный цвет B2E-ассистента на темном фоне */
exports.onDarkTextAccentB2E = 'var(--on-dark-text-accent-b2-e, #24B23E)';
/** Акцентный цвет бренда на темном фоне */
exports.onDarkTextAccentBrand = 'var(--on-dark-text-accent-brand, #528EFF)';
/** Градиент Афины на темном фоне */
exports.onDarkTextGradientAthena = 'var(--on-dark-text-gradient-athena, linear-gradient(93.97deg, #52BAFF 6.49%, #ADADFF 93.51%))';
/** Градиент Джой на темном фоне */
exports.onDarkTextGradientJoy = 'var(--on-dark-text-gradient-joy, linear-gradient(93.97deg, #FD9C68 6.49%, #F391E2 93.51%))';
/** Градиент B2E-ассистента на темном фоне */
exports.onDarkTextGradientB2E = 'var(--on-dark-text-gradient-b2-e, linear-gradient(89.83deg, #24B23E 0%, #1CC9C9 100%))';
/** Градиент бренда на темном фоне */
exports.onDarkTextGradientBrand = 'var(--on-dark-text-gradient-brand, linear-gradient(89.83deg, #528EFF 0%, #04C6C9 100%))';
/** Градиент на темном фоне */
exports.onDarkTextGradientMain = 'var(--on-dark-text-gradient-main, linear-gradient(89.83deg, #24B23E 0%, #1CC9C9 100%))';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccentMain = 'var(--on-dark-text-accent-main, #24B23E)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimary = 'var(--on-light-text-primary, rgba(8, 8, 8, 0.96))';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondary = 'var(--on-light-text-secondary, rgba(8, 8, 8, 0.56))';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiary = 'var(--on-light-text-tertiary, #08080847)';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraph = 'var(--on-light-text-paragraph, rgba(8, 8, 8, 0.8))';
/** Цвет успеха на светлом фоне */
exports.onLightTextPositive = 'var(--on-light-text-positive, #1A9E32)';
/** Цвет предупреждения на светлом фоне */
exports.onLightTextWarning = 'var(--on-light-text-warning, #FA5F05)';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegative = 'var(--on-light-text-negative, #FF293E)';
/** Акцентный цвет Афины на светлом фоне */
exports.onLightTextAccentAthena = 'var(--on-light-text-accent-athena, #118CDF)';
/** Акцентный цвет Джой на светлом фоне */
exports.onLightTextAccentJoy = 'var(--on-light-text-accent-joy, #B559F3)';
/** Акцентный цвет B2E-ассистента на светлом фоне */
exports.onLightTextAccentB2E = 'var(--on-light-text-accent-b2-e, #1A9E32)';
/** Акцентный цвет бренда на светлом фоне */
exports.onLightTextAccentBrand = 'var(--on-light-text-accent-brand, #3F81FD)';
/** Градиент Афины на светлом фоне */
exports.onLightTextGradientAthena = 'var(--on-light-text-gradient-athena, linear-gradient(89.83deg, #118CDF 0%, #7A7AFF 100%))';
/** Градиент Джой на светлом фоне */
exports.onLightTextGradientJoy = 'var(--on-light-text-gradient-joy, linear-gradient(89.83deg, #FCA349 0%, #FF52C5 100%))';
/** Градиент B2E-ассистента на светлом фоне */
exports.onLightTextGradientB2E = 'var(--on-light-text-gradient-b2-e, linear-gradient(89.83deg, #24B23E 0%, #00A4D1 100%))';
/** Акцентный цвет бренда на светлом фоне */
exports.onLightTextGradientBrand = 'var(--on-light-text-gradient-brand, linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%))';
/** Градиент на светлом фоне */
exports.onLightTextGradientMain = 'var(--on-light-text-gradient-main, linear-gradient(89.83deg, #24B23E 0%, #00A4D1 100%))';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccentMain = 'var(--on-light-text-accent-main, #1A9E32)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimary = 'var(--inverse-text-primary, rgba(8, 8, 8, 0.96))';
/** Инвертированный вторичный цвет текста */
exports.inverseTextSecondary = 'var(--inverse-text-secondary, rgba(8, 8, 8, 0.56))';
/** Инвертированный третичный цвет текста */
exports.inverseTextTertiary = 'var(--inverse-text-tertiary, #08080847)';
/** Инвертированный cплошной наборный текст */
exports.inverseTextParagraph = 'var(--inverse-text-paragraph, rgba(8, 8, 8, 0.8))';
/** Непрозрачный фон поверхности/контрола по умолчанию */
exports.surfaceSolidDefault = 'var(--surface-solid-default, #FFFFFF)';
/** Основной непрозрачный фон поверхности/контрола */
exports.surfaceSolidPrimary = 'var(--surface-solid-primary, #171717)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondary = 'var(--surface-solid-secondary, #262626)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiary = 'var(--surface-solid-tertiary, #363636)';
/** Основной фон для карточек */
exports.surfaceSolidCard = 'var(--surface-solid-card, #171717)';
/** Основной прозрачный фон поверхности/контрола */
exports.surfaceTransparentPrimary = 'var(--surface-transparent-primary, rgba(255, 255, 255, 0.06))';
/** Вторичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentSecondary = 'var(--surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentTertiary = 'var(--surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный фон для карточек */
exports.surfaceTransparentCard = 'var(--surface-transparent-card, rgba(255,255,255,0.06))';
/** Цвет фона поверхности/контрола успех */
exports.surfacePositive = 'var(--surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола предупреждение */
exports.surfaceWarning = 'var(--surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegative = 'var(--surface-negative, #FF293E)';
/** Фон поверхности/контрола без заливки */
exports.surfaceClear = 'var(--surface-clear, rgba(0, 0, 0, 0))';
/** Акцентный цвет поверхности/контролла Афины */
exports.surfaceAccentAthena = 'var(--surface-accent-athena, #118CDF)';
/** Акцентный цвет поверхности/контролла Джой */
exports.surfaceAccentJoy = 'var(--surface-accent-joy, #B559F3)';
/** Акцентный цвет поверхности/контролла B2E-ассистента */
exports.surfaceAccentB2E = 'var(--surface-accent-b2-e, #1A9E32)';
/** Акцентный цвет поверхности/контролла бренда */
exports.surfaceAccentBrand = 'var(--surface-accent-brand, #3F81FD)';
/** Цвет баббла сообщения от пользователя */
exports.surfaceSpeechBubbleSent = 'var(--surface-speech-bubble-sent, #08080847)';
/** Цвет баббла сообщения от ассистента */
exports.surfaceSpeechBubbleReceived = 'var(--surface-speech-bubble-received, #FFFFFF0F)';
/** Фон для скелетона */
exports.surfaceSkeleton = 'var(--surface-skeleton, linear-gradient(90deg, rgba(255, 255, 255, 0.00) 0%, rgba(255, 255, 255, 0.01) 17.19%, rgba(255, 255, 255, 0.05) 40.24%, rgba(255, 255, 255, 0.06) 51.17%, rgba(255, 255, 255, 0.05) 62.12%, rgba(255, 255, 255, 0.01) 82.81%, rgba(255, 255, 255, 0.00) 100%))';
/** Акцентный цвет поверхности/контролла */
exports.surfaceAccentMain = 'var(--surface-accent-main, #1A9E32)';
exports.surfaceGradientMain = 'var(--surface-gradient-main, linear-gradient(89.83deg, #24B23E 0%, #00A4D1 100%))';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
exports.onDarkSurfaceSolidDefault = 'var(--on-dark-surface-solid-default, #FFFFFF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidPrimary = 'var(--on-dark-surface-solid-primary, #171717)';
/** Вторичный непрозрачный фон поверхности/контрола на темном */
exports.onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #262626)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiary = 'var(--on-dark-surface-solid-tertiary, #363636)';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCard = 'var(--on-dark-surface-solid-card, #171717)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentPrimary = 'var(--on-dark-surface-transparent-primary, rgba(255, 255, 255, 0.06))';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentSecondary = 'var(--on-dark-surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentTertiary = 'var(--on-dark-surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный фон для карточек на темном фоне */
exports.onDarkSurfaceTransparentCard = 'var(--on-dark-surface-transparent-card, rgba(255,255,255,0.06))';
/** Цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfacePositive = 'var(--on-dark-surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола предупреждение на темном */
exports.onDarkSurfaceWarning = 'var(--on-dark-surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка на темном */
exports.onDarkSurfaceNegative = 'var(--on-dark-surface-negative, #FF293E)';
/** Акцентный цвет поверхности/контролла Афины */
exports.onDarkSurfaceAccentAthena = 'var(--on-dark-surface-accent-athena, #118CDF)';
/** Акцентный цвет поверхности/контролла Джой */
exports.onDarkSurfaceAccentJoy = 'var(--on-dark-surface-accent-joy, #B559F3)';
/** Акцентный цвет поверхности/контролла B2E-ассистента */
exports.onDarkSurfaceAccentB2E = 'var(--on-dark-surface-accent-b2-e, #1A9E32)';
/** Акцентный цвет поверхности/контролла бренда */
exports.onDarkSurfaceAccentBrand = 'var(--on-dark-surface-accent-brand, #3F81FD)';
/** Акцентный цвет поверхности/контролла */
exports.onDarkSurfaceAccentMain = 'var(--on-dark-surface-accent-main, #1A9E32)';
exports.onDarkSurfaceGradientMain = 'var(--on-dark-surface-gradient-main, linear-gradient(89.83deg, #24B23E 0%, #00A4D1 100%))';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
exports.onLightSurfaceSolidDefault = 'var(--on-light-surface-solid-default, #080808)';
/** Основной непрозрачный фон поверхности/контрола на светлом */
exports.onLightSurfaceSolidPrimary = 'var(--on-light-surface-solid-primary, #F5F5F5)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом */
exports.onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #ECECEC)';
/** Третичный непрозрачный фон поверхности/контрола на светлом */
exports.onLightSurfaceSolidTertiary = 'var(--on-light-surface-solid-tertiary, #DDDDDD)';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCard = 'var(--on-light-surface-solid-card, #FFFFFFFF)';
/** Основной прозрачный фон поверхности/контрола на светлом */
exports.onLightSurfaceTransparentPrimary = 'var(--on-light-surface-transparent-primary, #08080808)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentSecondary = 'var(--on-light-surface-transparent-secondary, rgba(8, 8, 8, 0.06))';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentTertiary = 'var(--on-light-surface-transparent-tertiary, rgba(8, 8, 8, 0.12))';
/** Прозрачный фон для карточек на светлом фоне */
exports.onLightSurfaceTransparentCard = 'var(--on-light-surface-transparent-card, #FFFFFFFF)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfacePositive = 'var(--on-light-surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола предупреждение на светлом */
exports.onLightSurfaceWarning = 'var(--on-light-surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegative = 'var(--on-light-surface-negative, #FF293E)';
/** Акцентный цвет поверхности/контролла Афины на светлом фоне */
exports.onLightSurfaceAccentAthena = 'var(--on-light-surface-accent-athena, #118CDF)';
/** Акцентный цвет поверхности/контролла Джой на светлом */
exports.onLightSurfaceAccentJoy = 'var(--on-light-surface-accent-joy, #B559F3)';
/** Акцентный цвет поверхности B2E-ассистента на светлом */
exports.onLightSurfaceAccentB2E = 'var(--on-light-surface-accent-b2-e, #1A9E32)';
/** Акцентный цвет поверхности/контролла бренда на светлом */
exports.onLightSurfaceAccentBrand = 'var(--on-light-surface-accent-brand, #3F81FD)';
/** Прозрачный цвет поверхности/контролла на светлом фоне */
exports.onLightSurfaceTransparentDeep = 'var(--on-light-surface-transparent-deep, #080808A2)';
/** Акцентный цвет поверхности/контролла на светлом фоне */
exports.onLightSurfaceAccentMain = 'var(--on-light-surface-accent-main, #1A9E32)';
exports.onLightSurfaceGradientMain = 'var(--on-light-surface-gradient-main, linear-gradient(89.83deg, #24B23E 0%, #00A4D1 100%))';
/** Первичный фон */
exports.backgroundPrimary = 'var(--background-primary, #000000FF)';
/** Вторичный фон */
exports.backgroundSecondary = 'var(--background-secondary, #171717)';
/** Фон для модальной шторки */
exports.backgroundModalSheet = 'var(--background-modal-sheet, linear-gradient(180deg, rgba(255, 255, 255, 0.1) 0%, rgba(1, 182, 255, 0) 50%)), var(--background-modal-sheet, #080808FF)';
/** Фон Афины на мобиле */
exports.backgroundMobileAssistantAthena = 'var(--background-mobile-assistant-athena, linear-gradient(336.9deg, #143787 0%, rgba(8, 8, 8, 0) 64.95%)), var(--background-mobile-assistant-athena, radial-gradient(66.53% 100% at 73.33% 100%, rgba(110, 5, 193, 0.44) 0%, rgba(53, 19, 149, 0.1) 99.69%)), var(--background-mobile-assistant-athena, linear-gradient(180.03deg, rgba(8, 8, 8, 0) 50%, rgba(25, 63, 152, 0.41) 99.97%)), var(--background-mobile-assistant-athena, linear-gradient(270deg, rgba(39, 15, 107, 0.3) 0%, rgba(8, 8, 8, 0) 100%)), var(--background-mobile-assistant-athena, #080808FF)';
/** Фон Джой на мобиле */
exports.backgroundMobileAssistantJoy = 'var(--background-mobile-assistant-joy, linear-gradient(336.9deg, rgba(255, 156, 101, 0.24) 0%, rgba(8, 8, 8, 0) 64.95%)), var(--background-mobile-assistant-joy, radial-gradient(66.53% 100% at 73.33% 100%, rgba(61, 19, 149, 0.34) 0%, rgba(19, 24, 149, 0.1) 99.69%)), var(--background-mobile-assistant-joy, linear-gradient(180.03deg, rgba(8, 8, 8, 0) 50%, rgba(255, 215, 179, 0.15) 99.97%)), var(--background-mobile-assistant-joy, linear-gradient(270deg, rgba(107, 15, 87, 0.2) 0%, rgba(8, 8, 8, 0) 100%)), var(--background-mobile-assistant-joy, #080808FF)';
/** Фоновый градиент B2E-ассистента на мобиле */
exports.backgroundMobileAssistantB2E = 'var(--background-mobile-assistant-b2-e, linear-gradient(336.84deg, rgba(20, 116, 70, 0.6) 0%, rgba(8, 8, 8, 0) 64.88%)), var(--background-mobile-assistant-b2-e, radial-gradient(100% 100% at 75.89% 100%, rgba(0, 133, 255, 0.24) 0%, rgba(0, 71, 255, 0.03) 100%)), var(--background-mobile-assistant-b2-e, linear-gradient(180deg, rgba(8, 8, 8, 0) 50%, rgba(7, 71, 33, 0.3) 100%)), var(--background-mobile-assistant-b2-e, linear-gradient(270deg, #061621 0%, rgba(8, 8, 8, 0) 100%)), var(--background-mobile-assistant-b2-e, #080808FF)';
/** Фоновый градиент бренда на мобиле */
exports.backgroundMobileAssistantBrand = 'var(--background-mobile-assistant-brand, linear-gradient(336.9deg, #184BAB 0%, rgba(8, 8, 8, 0) 64.95%)), var(--background-mobile-assistant-brand, radial-gradient(66.53% 100% at 73.33% 100%, rgba(22, 69, 164, 0.44) 0%, rgba(11, 39, 45, 0.1) 100%)), var(--background-mobile-assistant-brand, linear-gradient(180.03deg, rgba(8, 8, 8, 0) 50%, rgba(22, 69, 164, 0.41) 99.97%)), var(--background-mobile-assistant-brand, linear-gradient(269.47deg, rgba(22, 69, 164, 0.3) 0.52%, rgba(8, 8, 8, 0) 99.61%)), var(--background-mobile-assistant-brand, #080808FF)';
/** Фон ассистента на мобиле */
exports.backgroundMobileAssistantMain = 'var(--background-mobile-assistant-main, linear-gradient(336.84deg, rgba(20, 116, 70, 0.6) 0%, rgba(8, 8, 8, 0) 64.88%)), var(--background-mobile-assistant-main, radial-gradient(100% 100% at 75.89% 100%, rgba(0, 133, 255, 0.24) 0%, rgba(0, 71, 255, 0.03) 100%)), var(--background-mobile-assistant-main, linear-gradient(180deg, rgba(8, 8, 8, 0) 50%, rgba(7, 71, 33, 0.3) 100%)), var(--background-mobile-assistant-main, linear-gradient(270deg, #061621 0%, rgba(8, 8, 8, 0) 100%)), var(--background-mobile-assistant-main, #080808FF)';
exports.skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
exports.skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** @deprecated instead use onDarkTextPrimary */
exports.whitePrimary = 'var(--plasma-colors-white-primary, var(--on-dark-text-primary))';
/** @deprecated instead use onDarkTextSecondary */
exports.whiteSecondary = 'var(--plasma-colors-white-secondary, var(--on-dark-text-secondary))';
/** @deprecated instead use onDarkTextTertiary */
exports.whiteTertiary = 'var(--plasma-colors-white-tertiary, var(--on-dark-text-tertiary))';
/** @deprecated instead use onLightTextPrimary */
exports.blackPrimary = 'var(--plasma-colors-black-primary, var(--on-light-text-primary))';
/** @deprecated instead use onLightTextSecondary */
exports.blackSecondary = 'var(--plasma-colors-black-secondary, var(--on-light-text-secondary))';
/** @deprecated instead use onLightTextTertiary */
exports.blackTertiary = 'var(--plasma-colors-black-tertiary, var(--on-light-text-tertiary))';
/** @deprecated instead use onLightSurfaceSolidDefault */
exports.buttonBlack = 'var(--plasma-colors-button-black, var(--on-light-surface-solid-default))';
/** @deprecated instead use onLightSurfaceTransparentSecondary */
exports.buttonBlackSecondary = 'var(--plasma-colors-button-black-secondary, var(--on-light-surface-transparent-secondary))';
/** @deprecated instead use onDarkSurfaceSolidDefault */
exports.buttonWhite = 'var(--plasma-colors-button-white, var(--on-dark-surface-solid-default))';
/** @deprecated instead use onDarkSurfaceTransparentSecondary */
exports.buttonWhiteSecondary = 'var(--plasma-colors-button-white-secondary, var(--on-dark-surface-transparent-secondary))';
/** @deprecated instead use textPrimary */
exports.text = 'var(--plasma-colors-text, var(--text-primary))';
/** @deprecated instead use textPrimary */
exports.primary = 'var(--plasma-colors-primary, var(--text-primary))';
/** @deprecated instead use textSecondary */
exports.secondary = 'var(--plasma-colors-secondary, var(--text-secondary))';
/** @deprecated instead use textTertiary */
exports.tertiary = 'var(--plasma-colors-tertiary, var(--text-tertiary))';
/** @deprecated instead use textParagraph */
exports.paragraph = 'var(--plasma-colors-paragraph, var(--text-paragraph))';
/** @deprecated instead use backgroundPrimary */
exports.background = 'var(--plasma-colors-background, var(--background-primary))';
/** @deprecated instead use textPositive */
exports.success = 'var(--plasma-colors-success, var(--text-positive))';
/** @deprecated instead use textWarning */
exports.warning = 'var(--plasma-colors-warning, var(--text-warning))';
/** @deprecated instead use textNegative */
exports.critical = 'var(--plasma-colors-critical, var(--text-negative))';
/** @deprecated instead use surfaceTransparentPrimary */
exports.surfaceLiquid01 = 'var(--plasma-colors-surface-liquid01, var(--surface-transparent-primary))';
/** @deprecated instead use surfaceTransparentSecondary */
exports.surfaceLiquid02 = 'var(--plasma-colors-surface-liquid02, var(--surface-transparent-secondary))';
/** @deprecated instead use surfaceTransparentTertiary */
exports.surfaceLiquid03 = 'var(--plasma-colors-surface-liquid03, var(--surface-transparent-tertiary))';
/** @deprecated instead use surfaceSolidPrimary */
exports.surfaceSolid01 = 'var(--plasma-colors-surface-solid01, var(--surface-solid-primary))';
/** @deprecated instead use surfaceSolidSecondary */
exports.surfaceSolid02 = 'var(--plasma-colors-surface-solid02, var(--surface-solid-secondary))';
/** @deprecated instead use surfaceSolidTertiary */
exports.surfaceSolid03 = 'var(--plasma-colors-surface-solid03, var(--surface-solid-tertiary))';
/** @deprecated instead use surfaceTransparentCard */
exports.surfaceCard = 'var(--plasma-colors-surface-card, var(--surface-transparent-card))';
/** @deprecated instead use surfaceTransparentSecondary */
exports.buttonSecondary = 'var(--plasma-colors-button-secondary, var(--surface-transparent-secondary))';
/** @deprecated instead use surfacePositive */
exports.buttonSuccess = 'var(--plasma-colors-button-success, var(--surface-positive))';
/** @deprecated instead use surfaceWarning */
exports.buttonWarning = 'var(--plasma-colors-button-warning, var(--surface-warning))';
/** @deprecated instead use surfaceNegative */
exports.buttonCritical = 'var(--plasma-colors-button-critical, var(--surface-negative))';
