// @sberdevices/plasma-web/es/components/Progress/Progress.js
import React from 'react';
import styled, { css } from 'styled-components';
import { caption, tertiary, surfaceLiquid02 } from '@sberdevices/plasma-core';
var statuses = {
  success: /*#__PURE__*/css(["background-color:#09a552;"]),
  warning: /*#__PURE__*/css(["background-color:#ee6820;"]),
  error: /*#__PURE__*/css(["background-color:#df2638;"])
};
var StyledRoot = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-web__sc-1e491uq-0"
})(["display:flex;align-items:center;"]);
var StyledTrack = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-web__sc-1e491uq-1"
})(["width:100%;height:0.25rem;background-color:", ";border-radius:0.125rem;"], surfaceLiquid02);
var StyledProgress = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-web__sc-1e491uq-2"
})(["width:", "%;height:100%;", " border-radius:0.125rem;"], function (_ref) {
  var value = _ref.value;
  return value;
}, function (_ref2) {
  var status = _ref2.status;
  return status && statuses[status];
});
var StyledValue = /*#__PURE__*/styled.span.withConfig({
  componentId: "plasma-web__sc-1e491uq-3"
})(["", ";margin-left:0.5rem;color:", ";"], caption, tertiary);
/**
 * Компонент для отображения прогресса в процентах.
 */

export var Progress = function Progress(_ref3) {
  var value = _ref3.value,
      status = _ref3.status,
      _ref3$displayValue = _ref3.displayValue,
      displayValue = _ref3$displayValue === void 0 ? true : _ref3$displayValue;
  var min = 0;
  var max = 100;
  var normalizedValue = Math.max(Math.min(value, max), min);
  return /*#__PURE__*/React.createElement(StyledRoot, null, /*#__PURE__*/React.createElement(StyledTrack, null, /*#__PURE__*/React.createElement(StyledProgress, {
    value: normalizedValue,
    status: status
  })), displayValue && /*#__PURE__*/React.createElement(StyledValue, null, normalizedValue, "%"));
};
