// @salutejs/plasma-tokens/brands/sberonline/index.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.buttonCritical = exports.buttonWarning = exports.buttonSuccess = exports.buttonAccent = exports.buttonSecondary = exports.surfaceCard = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.surfaceSolid01 = exports.surfaceLiquid03 = exports.surfaceLiquid02 = exports.surfaceLiquid01 = exports.overlay = exports.critical = exports.warning = exports.success = exports.accent = exports.background = exports.paragraph = exports.tertiary = exports.secondary = exports.primary = exports.text = exports.buttonWhiteSecondary = exports.buttonWhite = exports.buttonBlackSecondary = exports.buttonBlack = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.skeletonGradientLighter = exports.skeletonGradient = exports.inverseBackgroundPrimary = exports.lightBackgroundPrimary = exports.darkBackgroundPrimary = exports.backgroundPrimary = exports.inverseSurfaceTransparentDeep = exports.inverseSurfaceClear = exports.inverseSurfaceNegative = exports.inverseSurfaceWarning = exports.inverseSurfacePositive = exports.inverseSurfaceAccentGradient = exports.inverseSurfaceAccent = exports.inverseSurfaceTransparentCard = exports.inverseSurfaceTransparentTertiary = exports.inverseSurfaceTransparentSecondary = exports.inverseSurfaceTransparentPrimary = exports.inverseSurfaceSolidCard = exports.inverseSurfaceSolidTertiary = exports.inverseSurfaceSolidSecondary = exports.inverseSurfaceSolidPrimary = exports.inverseSurfaceSolidDefault = exports.onLightSurfaceTransparentDeep = exports.onLightSurfaceClear = exports.onLightSurfaceNegative = exports.onLightSurfaceWarning = exports.onLightSurfacePositive = exports.onLightSurfaceAccentGradient = exports.onLightSurfaceAccent = exports.onLightSurfaceTransparentCard = exports.onLightSurfaceTransparentTertiary = exports.onLightSurfaceTransparentSecondary = exports.onLightSurfaceTransparentPrimary = exports.onLightSurfaceSolidCard = exports.onLightSurfaceSolidTertiary = exports.onLightSurfaceSolidSecondary = exports.onLightSurfaceSolidPrimary = exports.onLightSurfaceSolidDefault = exports.onDarkSurfaceTransparentDeep = exports.onDarkSurfaceClear = exports.onDarkSurfaceNegative = exports.onDarkSurfaceWarning = exports.onDarkSurfacePositive = exports.onDarkSurfaceAccentGradient = exports.onDarkSurfaceAccent = exports.onDarkSurfaceTransparentCard = exports.onDarkSurfaceTransparentTertiary = exports.onDarkSurfaceTransparentSecondary = exports.onDarkSurfaceTransparentPrimary = exports.onDarkSurfaceSolidCard = exports.onDarkSurfaceSolidTertiary = exports.onDarkSurfaceSolidSecondary = exports.onDarkSurfaceSolidPrimary = exports.onDarkSurfaceSolidDefault = exports.surfaceTransparentDeep = exports.surfaceClear = exports.surfaceNegative = exports.surfaceWarning = exports.surfacePositive = exports.surfaceAccentGradient = exports.surfaceAccent = exports.surfaceTransparentCard = exports.surfaceTransparentTertiary = exports.surfaceTransparentSecondary = exports.surfaceTransparentPrimary = exports.surfaceSolidCard = exports.surfaceSolidTertiary = exports.surfaceSolidSecondary = exports.surfaceSolidPrimary = exports.surfaceSolidDefault = exports.inverseTextNegative = exports.inverseTextWarning = exports.inverseTextPositive = exports.inverseTextAccentGradient = exports.inverseTextAccent = exports.inverseTextParagraph = exports.inverseTextTertiary = exports.inverseTextSecondary = exports.inverseTextPrimary = exports.onLightTextNegative = exports.onLightTextWarning = exports.onLightTextPositive = exports.onLightTextAccentGradient = exports.onLightTextAccent = exports.onLightTextParagraph = exports.onLightTextTertiary = exports.onLightTextSecondary = exports.onLightTextPrimary = exports.onDarkTextNegative = exports.onDarkTextWarning = exports.onDarkTextPositive = exports.onDarkTextAccentGradient = exports.onDarkTextAccent = exports.onDarkTextParagraph = exports.onDarkTextTertiary = exports.onDarkTextSecondary = exports.onDarkTextPrimary = exports.textNegative = exports.textWarning = exports.textPositive = exports.textAccentGradient = exports.textAccent = exports.textParagraph = exports.textTertiary = exports.textSecondary = exports.textPrimary = exports.inverseOverlayHard = exports.inverseOverlaySoft = exports.onLightOverlayHard = exports.onLightOverlaySoft = exports.onDarkOverlayHard = exports.onDarkOverlaySoft = exports.overlayHard = exports.overlaySoft = void 0;
/** Цвет фона паранжи светлый */
exports.overlaySoft = 'var(--overlay-soft, #0808088F)';
/** Цвет фона паранжи темный */
exports.overlayHard = 'var(--overlay-hard, #080808f5)';
/** Цвет фона паранжи светлый на темном фоне */
exports.onDarkOverlaySoft = 'var(--on-dark-overlay-soft, #0808088f)';
/** Цвет фона паранжи темный на темном фоне */
exports.onDarkOverlayHard = 'var(--on-dark-overlay-hard, #080808f5)';
/** Цвет фона паранжи светлый на светлом фоне */
exports.onLightOverlaySoft = 'var(--on-light-overlay-soft, #f9f9f98f)';
/** Цвет фона паранжи темный на светлом фоне */
exports.onLightOverlayHard = 'var(--on-light-overlay-hard, #f9f9f9f5)';
/** Инвертированный цвет фона паранжи светлый */
exports.inverseOverlaySoft = 'var(--inverse-overlay-soft, #f9f9f98F)';
/** Инвертированный цвет фона паранжи темный */
exports.inverseOverlayHard = 'var(--inverse-overlay-hard, #f9f9f9f5)';
/** Основной цвет текста */
exports.textPrimary = 'var(--text-primary, #f5f5f5F5)';
/** Вторичный цвет текста */
exports.textSecondary = 'var(--text-secondary, #F5f5f58F)';
/** Третичный цвет текста */
exports.textTertiary = 'var(--text-tertiary, #F5f5f548)';
/** Сплошной наборный текст */
exports.textParagraph = 'var(--text-paragraph, #f5f5f5cc)';
/** Акцентный цвет */
exports.textAccent = 'var(--text-accent, #24B23E)';
/** Акцентный цвет с градиентом */
exports.textAccentGradient = 'var(--text-accent-gradient, linear-gradient(45deg, #1A9E32 0%, #04C6C9 100%))';
/** Цвет успеха */
exports.textPositive = 'var(--text-positive, #24B23E)';
/** Цвет предупреждения */
exports.textWarning = 'var(--text-warning, #FF7024)';
/** Цвет ошибки */
exports.textNegative = 'var(--text-negative, #FF3D51)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimary = 'var(--on-dark-text-primary, #f5f5f5F5)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondary = 'var(--on-dark-text-secondary, #F5f5f58F)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiary = 'var(--on-dark-text-tertiary, #F5f5f548)';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraph = 'var(--on-dark-text-paragraph, #f5f5f5CC)';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccent = 'var(--on-dark-text-accent, #24B23E)';
/** Акцентный цвет с градиентом на темном фоне */
exports.onDarkTextAccentGradient = 'var(--on-dark-text-accent-gradient, linear-gradient(45deg, #1A9E32 0%, #04C6C9 100%))';
/** Цвет успеха на темном фоне */
exports.onDarkTextPositive = 'var(--on-dark-text-positive, #24B23E)';
/** Цвет предупреждения на темном фоне */
exports.onDarkTextWarning = 'var(--on-dark-text-warning, #FF7024)';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegative = 'var(--on-dark-text-negative, #FF3D51)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimary = 'var(--on-light-text-primary, #171717F5)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondary = 'var(--on-light-text-secondary, #1717178F)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiary = 'var(--on-light-text-tertiary, #17171748)';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraph = 'var(--on-light-text-paragraph, #171717CC)';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccent = 'var(--on-light-text-accent, #108E26)';
/** Акцентный цвет с градиентом на светлом фоне */
exports.onLightTextAccentGradient = 'var(--on-light-text-accent-gradient, linear-gradient(45deg, #0D8523 0%, #0DA8AB 100%))';
/** Цвет успеха на светлом фоне */
exports.onLightTextPositive = 'var(--on-light-text-positive, #108E26)';
/** Цвет предупреждения на светлом фоне */
exports.onLightTextWarning = 'var(--on-light-text-warning, #E85702)';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegative = 'var(--on-light-text-negative, #F31B31)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimary = 'var(--inverse-text-primary, #171717F5)';
/** Инвертированный вторичный цвет текста */
exports.inverseTextSecondary = 'var(--inverse-text-secondary, #1717178F)';
/** Инвертированный третичный цвет текста */
exports.inverseTextTertiary = 'var(--inverse-text-tertiary, #17171748)';
/** Инвертированный cплошной наборный текст */
exports.inverseTextParagraph = 'var(--inverse-text-paragraph, #171717CC)';
/** Инвертированный акцентный цвет */
exports.inverseTextAccent = 'var(--inverse-text-accent, #108E26)';
/** Инвертированный акцентный цвет с градиентом */
exports.inverseTextAccentGradient = 'var(--inverse-text-accent-gradient, linear-gradient(45deg, #0D8523 0%, #0DA8AB 100%))';
/** Инвертированный цвет успеха */
exports.inverseTextPositive = 'var(--inverse-text-positive, #108E26)';
/** Инвертированный цвет предупреждения */
exports.inverseTextWarning = 'var(--inverse-text-warning, #E85702)';
/** Инвертированный цвет ошибки */
exports.inverseTextNegative = 'var(--inverse-text-negative, #F31B31)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
exports.surfaceSolidDefault = 'var(--surface-solid-default, #FFFFFFFF)';
/** Основной непрозрачный фон поверхности/контрола */
exports.surfaceSolidPrimary = 'var(--surface-solid-primary, #171717)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondary = 'var(--surface-solid-secondary, #262626)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiary = 'var(--surface-solid-tertiary, #363636)';
/** Основной фон для карточек */
exports.surfaceSolidCard = 'var(--surface-solid-card, #171717)';
/** Основной прозрачный фон поверхности/контрола */
exports.surfaceTransparentPrimary = 'var(--surface-transparent-primary, #FFFFFF0F)';
/** Вторичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentSecondary = 'var(--surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentTertiary = 'var(--surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный фон для карточек */
exports.surfaceTransparentCard = 'var(--surface-transparent-card, #FFFFFF0F)';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccent = 'var(--surface-accent, #1A9E32)';
/** Акцентный фон поверхности/контрола с градиентом */
exports.surfaceAccentGradient = 'var(--surface-accent-gradient, linear-gradient(45deg, #1A9E32 0%, #04C6C9 100%))';
/** Цвет фона поверхности/контрола успех */
exports.surfacePositive = 'var(--surface-positive, #1A9E32)';
/** Цвет фона поверхности предупреждение */
exports.surfaceWarning = 'var(--surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegative = 'var(--surface-negative, #FF293E)';
/** Фон поверхности/контрола без заливки */
exports.surfaceClear = 'var(--surface-clear, rgba(0, 0, 0, 0))';
/** Прозрачный фон поверхности/контрола по умолчанию */
exports.surfaceTransparentDeep = 'var(--surface-transparent-deep, #FFFFFFA3)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
exports.onDarkSurfaceSolidDefault = 'var(--on-dark-surface-solid-default, #FFFFFF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidPrimary = 'var(--on-dark-surface-solid-primary, #171717)';
/** Вторичный непрозрачный фон поверхности на темном фоне */
exports.onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #262626)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiary = 'var(--on-dark-surface-solid-tertiary, #363636)';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCard = 'var(--on-dark-surface-solid-card, #171717)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentPrimary = 'var(--on-dark-surface-transparent-primary, rgba(255, 255, 255, 0.06))';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentSecondary = 'var(--on-dark-surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentTertiary = 'var(--on-dark-surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный фон для карточек на темном фоне */
exports.onDarkSurfaceTransparentCard = 'var(--on-dark-surface-transparent-card, #FFFFFF0F)';
/** Акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccent = 'var(--on-dark-surface-accent, #1A9E32)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
exports.onDarkSurfaceAccentGradient = 'var(--on-dark-surface-accent-gradient, linear-gradient(45deg, #1A9E32 0%, #04C6C9 100%))';
/** Цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfacePositive = 'var(--on-dark-surface-positive, #1A9E32)';
/** Цвет фона поверхности предупреждение на темном фоне */
exports.onDarkSurfaceWarning = 'var(--on-dark-surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegative = 'var(--on-dark-surface-negative, #FF293E)';
/** Фон поверхности/контрола без заливки на темном фоне */
exports.onDarkSurfaceClear = 'var(--on-dark-surface-clear, rgba(0, 0, 0, 0))';
/** Прозрачный фон поверхности по умолчанию на темном фоне */
exports.onDarkSurfaceTransparentDeep = 'var(--on-dark-surface-transparent-deep, #FFFFFFA3)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
exports.onLightSurfaceSolidDefault = 'var(--on-light-surface-solid-default, #080808)';
/** Основной непрозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceSolidPrimary = 'var(--on-light-surface-solid-primary, #F5F5F5)';
/** Вторичный непрозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #ECECEC)';
/** Третичный непрозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceSolidTertiary = 'var(--on-light-surface-solid-tertiary, #DDDDDD)';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCard = 'var(--on-light-surface-solid-card, #FFFFFFFF)';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentPrimary = 'var(--on-light-surface-transparent-primary, rgba(8, 8, 8, 0.02))';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentSecondary = 'var(--on-light-surface-transparent-secondary, rgba(8, 8, 8, 0.06))';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentTertiary = 'var(--on-light-surface-transparent-tertiary, rgba(8, 8, 8, 0.12))';
/** Прозрачный фон для карточек на светлом фоне */
exports.onLightSurfaceTransparentCard = 'var(--on-light-surface-transparent-card, rgba(8, 8, 8, 0.06))';
/** Акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccent = 'var(--on-light-surface-accent, #1A9E32)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
exports.onLightSurfaceAccentGradient = 'var(--on-light-surface-accent-gradient, linear-gradient(45deg, #1A9E32 0%, #04C6C9 100%))';
/** Цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfacePositive = 'var(--on-light-surface-positive, #1A9E32)';
/** Цвет фона поверхности / предупреждение на светлом фоне */
exports.onLightSurfaceWarning = 'var(--on-light-surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegative = 'var(--on-light-surface-negative, #FF293E)';
/** Фон поверхности/контрола без заливки на светлом фоне */
exports.onLightSurfaceClear = 'var(--on-light-surface-clear, rgba(0, 0, 0, 0))';
/** Прозрачный фон поверхности по умолчанию на светлом фоне */
exports.onLightSurfaceTransparentDeep = 'var(--on-light-surface-transparent-deep, #080808A3)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
exports.inverseSurfaceSolidDefault = 'var(--inverse-surface-solid-default, #080808)';
/** Инвертированный основной непрозрачный фон поверхности */
exports.inverseSurfaceSolidPrimary = 'var(--inverse-surface-solid-primary, #F5F5F5)';
/** Инвертированный вторичный непрозрачный фон поверхности */
exports.inverseSurfaceSolidSecondary = 'var(--inverse-surface-solid-secondary, #ECECEC)';
/** Инвертированный третичный непрозрачный фон поверхности */
exports.inverseSurfaceSolidTertiary = 'var(--inverse-surface-solid-tertiary, #DDDDDD)';
/** Инвертированный основной фон для карточек */
exports.inverseSurfaceSolidCard = 'var(--inverse-surface-solid-card, #FFFFFFFF)';
/** Инвертированный основной прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentPrimary = 'var(--inverse-surface-transparent-primary, rgba(8, 8, 8, 0.02))';
/** Инвертированный вторичный прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentSecondary = 'var(--inverse-surface-transparent-secondary, rgba(8, 8, 8, 0.06))';
/** Инвертированный третичный прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentTertiary = 'var(--inverse-surface-transparent-tertiary, rgba(8, 8, 8, 0.12))';
/** Инвертированный прозрачный фон для карточек */
exports.inverseSurfaceTransparentCard = 'var(--inverse-surface-transparent-card, #FFFFFFFF)';
/** Инвертированный акцентный фон поверхности/контрола */
exports.inverseSurfaceAccent = 'var(--inverse-surface-accent, #1A9E32)';
/** Инвертированный акцентный фон поверхности/контрола с градиентом */
exports.inverseSurfaceAccentGradient = 'var(--inverse-surface-accent-gradient, linear-gradient(45deg, #1A9E32 0%, #04C6C9 100%))';
/** Инвертированный цвет фона поверхности/контрола успех */
exports.inverseSurfacePositive = 'var(--inverse-surface-positive, #1A9E32)';
/** Инвертированный цвет фона поверхности / предупреждение */
exports.inverseSurfaceWarning = 'var(--inverse-surface-warning, #FA5F05)';
/** Инвертированный цвет фона поверхности/контрола ошибка */
exports.inverseSurfaceNegative = 'var(--inverse-surface-negative, #FF293E)';
/** Инвертированный фон поверхности/контрола без заливки */
exports.inverseSurfaceClear = 'var(--inverse-surface-clear, rgba(0, 0, 0, 0))';
/** Инвертированный прозрачный фон поверхности по умолчанию */
exports.inverseSurfaceTransparentDeep = 'var(--inverse-surface-transparent-deep, #080808A3)';
/** Основной фон */
exports.backgroundPrimary = 'var(--background-primary, #080808)';
/** Основной фон на темном фоне */
exports.darkBackgroundPrimary = 'var(--dark-background-primary, #080808)';
/** Основной фон на светлом фоне */
exports.lightBackgroundPrimary = 'var(--light-background-primary, #F9F9F9)';
/** Инвертированный основной фон */
exports.inverseBackgroundPrimary = 'var(--inverse-background-primary, #F9F9F9)';
exports.skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
exports.skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** @deprecated instead use onDarkTextPrimary */
exports.whitePrimary = 'var(--plasma-colors-white-primary, var(--on-dark-text-primary))';
/** @deprecated instead use onDarkTextSecondary */
exports.whiteSecondary = 'var(--plasma-colors-white-secondary, var(--on-dark-text-secondary))';
/** @deprecated instead use onDarkTextTertiary */
exports.whiteTertiary = 'var(--plasma-colors-white-tertiary, var(--on-dark-text-tertiary))';
/** @deprecated instead use onLightTextPrimary */
exports.blackPrimary = 'var(--plasma-colors-black-primary, var(--on-light-text-primary))';
/** @deprecated instead use onLightTextSecondary */
exports.blackSecondary = 'var(--plasma-colors-black-secondary, var(--on-light-text-secondary))';
/** @deprecated instead use onLightTextTertiary */
exports.blackTertiary = 'var(--plasma-colors-black-tertiary, var(--on-light-text-tertiary))';
/** @deprecated instead use onLightSurfaceSolidDefault */
exports.buttonBlack = 'var(--plasma-colors-button-black, var(--on-light-surface-solid-default))';
/** @deprecated instead use onLightSurfaceTransparentSecondary */
exports.buttonBlackSecondary = 'var(--plasma-colors-button-black-secondary, var(--on-light-surface-transparent-secondary))';
/** @deprecated instead use onDarkSurfaceSolidDefault */
exports.buttonWhite = 'var(--plasma-colors-button-white, var(--on-dark-surface-solid-default))';
/** @deprecated instead use onDarkSurfaceTransparentSecondary */
exports.buttonWhiteSecondary = 'var(--plasma-colors-button-white-secondary, var(--on-dark-surface-transparent-secondary))';
/** @deprecated instead use textPrimary */
exports.text = 'var(--plasma-colors-text, var(--text-primary))';
/** @deprecated instead use textPrimary */
exports.primary = 'var(--plasma-colors-primary, var(--text-primary))';
/** @deprecated instead use textSecondary */
exports.secondary = 'var(--plasma-colors-secondary, var(--text-secondary))';
/** @deprecated instead use textTertiary */
exports.tertiary = 'var(--plasma-colors-tertiary, var(--text-tertiary))';
/** @deprecated instead use textParagraph */
exports.paragraph = 'var(--plasma-colors-paragraph, var(--text-paragraph))';
/** @deprecated instead use backgroundPrimary */
exports.background = 'var(--plasma-colors-background, var(--background-primary))';
/** @deprecated instead use textAccent */
exports.accent = 'var(--plasma-colors-accent, var(--text-accent))';
/** @deprecated instead use textPositive */
exports.success = 'var(--plasma-colors-success, var(--text-positive))';
/** @deprecated instead use textWarning */
exports.warning = 'var(--plasma-colors-warning, var(--text-warning))';
/** @deprecated instead use textNegative */
exports.critical = 'var(--plasma-colors-critical, var(--text-negative))';
/** @deprecated instead use overlaySoft */
exports.overlay = 'var(--plasma-colors-overlay, var(--overlay-soft))';
/** @deprecated instead use surfaceTransparentPrimary */
exports.surfaceLiquid01 = 'var(--plasma-colors-surface-liquid01, var(--surface-transparent-primary))';
/** @deprecated instead use surfaceTransparentSecondary */
exports.surfaceLiquid02 = 'var(--plasma-colors-surface-liquid02, var(--surface-transparent-secondary))';
/** @deprecated instead use surfaceTransparentTertiary */
exports.surfaceLiquid03 = 'var(--plasma-colors-surface-liquid03, var(--surface-transparent-tertiary))';
/** @deprecated instead use surfaceSolidPrimary */
exports.surfaceSolid01 = 'var(--plasma-colors-surface-solid01, var(--surface-solid-primary))';
/** @deprecated instead use surfaceSolidSecondary */
exports.surfaceSolid02 = 'var(--plasma-colors-surface-solid02, var(--surface-solid-secondary))';
/** @deprecated instead use surfaceSolidTertiary */
exports.surfaceSolid03 = 'var(--plasma-colors-surface-solid03, var(--surface-solid-tertiary))';
/** @deprecated instead use surfaceTransparentCard */
exports.surfaceCard = 'var(--plasma-colors-surface-card, var(--surface-transparent-card))';
/** @deprecated instead use surfaceTransparentSecondary */
exports.buttonSecondary = 'var(--plasma-colors-button-secondary, var(--surface-transparent-secondary))';
/** @deprecated instead use textAccent */
exports.buttonAccent = 'var(--plasma-colors-button-accent, var(--text-accent))';
/** @deprecated instead use surfacePositive */
exports.buttonSuccess = 'var(--plasma-colors-button-success, var(--surface-positive))';
/** @deprecated instead use surfaceWarning */
exports.buttonWarning = 'var(--plasma-colors-button-warning, var(--surface-warning))';
/** @deprecated instead use surfaceNegative */
exports.buttonCritical = 'var(--plasma-colors-button-critical, var(--surface-negative))';
