// @sberdevices/plasma-web/components/Select/SelectButton.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SelectButton = void 0;

var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));

var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));

var _plasmaCore = /*#__PURE__*/require("@sberdevices/plasma-core");

var _plasmaIcons = /*#__PURE__*/require("@sberdevices/plasma-icons");

var _tokens = /*#__PURE__*/require("../../tokens");

var _excluded = ["value", "placeholder", "hasItems"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var statuses = {
  success: _plasmaCore.success,
  warning: _plasmaCore.warning,
  error: _plasmaCore.critical
};
var StyledArrow = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaIcons.IconChevronDown).withConfig({
  componentId: "plasma-web__sc-ipkhs8-0"
})(["margin:var(--plasma-select-button-arrow-margin,0 0 0 0.75rem);transition:color 0.3s ease-in-out,transform 0.15s ease-in-out;pointer-events:none;user-select:none;"]);

var StyledText = /*#__PURE__*/_styledComponents["default"].span.withConfig({
  componentId: "plasma-web__sc-ipkhs8-1"
})(["", " color:", ";transition:color 0.3s ease-in-out;pointer-events:none;user-select:none;"], _plasmaCore.applyEllipsis, _plasmaCore.primary);

var StyledPlaceholder = /*#__PURE__*/_styledComponents["default"].span.withConfig({
  componentId: "plasma-web__sc-ipkhs8-2"
})(["", " color:", ";pointer-events:none;user-select:none;"], _plasmaCore.applyEllipsis, _plasmaCore.tertiary);

var StyledButton = /*#__PURE__*/_styledComponents["default"].button.withConfig({
  componentId: "plasma-web__sc-ipkhs8-3"
})(["", " display:flex;align-items:center;justify-content:space-between;box-sizing:border-box;width:100%;height:3rem;padding:0.875rem 0.9375rem;border:1px solid ", ";border-radius:0.25rem;background-color:transparent;color:", ";transition:border-color 0.3s ease-in-out;", " &:disabled{cursor:inherit;}&:not(:disabled):focus{outline:0 none;border-color:", ";color:", ";}", " ", " ", ""], _plasmaCore.body1, _tokens.inputBorder, _plasmaCore.secondary, function (_ref) {
  var hasItems = _ref.hasItems;
  return hasItems && (0, _styledComponents.css)(["&:not(:disabled):hover{cursor:pointer;border-color:", ";color:", ";}"], _tokens.inputBorderHover, _plasmaCore.secondary);
}, _plasmaCore.accent, _plasmaCore.accent, function (_ref2) {
  var focused = _ref2.focused;
  return focused && (0, _styledComponents.css)(["border-color:", ";color:", ";"], _plasmaCore.accent, _plasmaCore.accent);
}, function (_ref3) {
  var status = _ref3.status;
  return status && (0, _styledComponents.css)(["border-color:", ";color:", ";&:hover,&:focus{border-color:", ";}"], statuses[status], statuses[status], statuses[status]);
}, function (_ref4) {
  var isOpen = _ref4.isOpen;
  return isOpen && (0, _styledComponents.css)(["", "{transform:rotate(-180deg);}"], StyledArrow);
}); // eslint-disable-next-line prefer-arrow-callback


var SelectButton = /*#__PURE__*/(0, _react.forwardRef)(function SelectButton(_ref5, ref) {
  var value = _ref5.value,
      placeholder = _ref5.placeholder,
      hasItems = _ref5.hasItems,
      rest = _objectWithoutProperties(_ref5, _excluded);

  return /*#__PURE__*/_react["default"].createElement(StyledButton, _extends({}, rest, {
    ref: ref,
    hasItems: hasItems,
    "aria-label": placeholder
  }), value && /*#__PURE__*/_react["default"].createElement(StyledText, null, value), placeholder && !value && /*#__PURE__*/_react["default"].createElement(StyledPlaceholder, {
    "aria-hidden": "true"
  }, placeholder), hasItems && /*#__PURE__*/_react["default"].createElement(StyledArrow, {
    size: "xs",
    color: "inherit"
  }));
});
exports.SelectButton = SelectButton;
