// @sberdevices/plasma-ui/components/Button/Button.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Button = void 0;

var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));

var _plasmaCore = /*#__PURE__*/require("@sberdevices/plasma-core");

var _mixins = /*#__PURE__*/require("../../mixins");

var _Button = /*#__PURE__*/require("./Button.mixins");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var StyledButtonRoot = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaCore.ButtonRoot).withConfig({
  componentId: "plasma-ui__sc-1afi2fp-0"
})(["", " ", " ", " ", ""], _Button.applySizes, _Button.applyViews, _mixins.applyInteraction, _plasmaCore.applyNoSelect);
/**
 * Основной компонент для отображения кнопок.
 * Поддерживает несколько режимов отображения (`view`) и размеров (`size`).
 */

var Button = /*#__PURE__*/(0, _plasmaCore.createButton)(StyledButtonRoot);
exports.Button = Button;
Button.defaultProps = {
  size: 'l',
  view: 'secondary',
  pin: 'square-square',
  outlined: true,
  scaleOnInteraction: true
};
