// @sberdevices/plasma-core/components/Button/Button.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Button = exports.ButtonText = exports.ButtonRoot = void 0;

var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));

var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));

var _mixins = /*#__PURE__*/require("../../mixins");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

/**
 * Корневой элемент для стилизации кнопки.
 * Должен быть передан в функцию `createButton`.
 */
var ButtonRoot = /*#__PURE__*/_styledComponents["default"].button.withConfig({
  componentId: "plasma__sc-dcpyn7-0"
})(["position:relative;display:inline-flex;align-items:center;box-sizing:border-box;justify-content:center;appearance:none;border:none;cursor:pointer;-webkit-tap-highlight-color:rgba(0,0,0,0);a&{text-decoration:none;}&:focus{outline:none;}", " ", ""], _mixins.applyDisabled, _mixins.applyBlur);
/**
 * A text wrapping element, that used when `text` prop is passed.
 * @private
 */


exports.ButtonRoot = ButtonRoot;

var ButtonText = /*#__PURE__*/_styledComponents["default"].span.withConfig({
  componentId: "plasma__sc-dcpyn7-1"
})(["box-sizing:border-box;", " ", " ", ""], _mixins.applyEllipsis, function (_ref) {
  var isContentLeft = _ref.$isContentLeft;
  return isContentLeft && 'margin-left: 0.375rem;';
}, function (_ref2) {
  var isContentRight = _ref2.$isContentRight;
  return isContentRight && 'margin-right: 0.375rem;';
});
/**
 * Базовая кнопка.
 * @deprecated
 */


exports.ButtonText = ButtonText;

var Button = /*#__PURE__*/_react["default"].forwardRef( // eslint-disable-next-line prefer-arrow-callback
function Button(props, ref) {
  return /*#__PURE__*/_react["default"].createElement(ButtonRoot, _extends({
    ref: ref
  }, props));
});

exports.Button = Button;
Button.defaultProps = {
  pin: 'square-square',
  outlined: false
};
