// @sberdevices/plasma-web/tokens/colors.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.linkVisitedActive = exports.linkVisitedHover = exports.linkVisited = exports.linkActive = exports.linkHover = exports.link = exports.inputLabelColor = exports.inputErrorBorderColor = exports.inputErrorCaretColor = exports.inputErrorColor = exports.inputErrorBackground = exports.inputWarningBorderColor = exports.inputWarningCaretColor = exports.inputWarningColor = exports.inputWarningBackground = exports.inputSuccessBorderColor = exports.inputSuccessCaretColor = exports.inputSuccessColor = exports.inputSuccessBackground = exports.inputBorderFocus = exports.inputBorderHover = exports.inputBorder = exports.inputFocusPlaceholderColor = exports.inputPlaceholderColor = exports.inputCaretColor = exports.inputColor = exports.inputBackgroundFocus = exports.inputBackgroundHover = exports.inputBackground = exports.fieldLabelColor = exports.fieldColor = exports.buttonCheckedActiveColor = exports.buttonCheckedActive = exports.buttonCheckedHoverColor = exports.buttonCheckedHover = exports.buttonCriticalActive = exports.buttonCriticalHover = exports.buttonWarningActive = exports.buttonWarningHover = exports.buttonSuccessActive = exports.buttonSuccessHover = exports.buttonSecondaryActive = exports.buttonSecondaryHover = exports.buttonPrimaryActive = exports.buttonPrimaryHover = void 0;
var buttonPrimaryHover = 'var(--plasma-colors-button-primary-hover)';
exports.buttonPrimaryHover = buttonPrimaryHover;
var buttonPrimaryActive = 'var(--plasma-colors-button-primary-active)';
exports.buttonPrimaryActive = buttonPrimaryActive;
var buttonSecondaryHover = 'var(--plasma-colors-button-secondary-hover)';
exports.buttonSecondaryHover = buttonSecondaryHover;
var buttonSecondaryActive = 'var(--plasma-colors-button-secondary-active)';
exports.buttonSecondaryActive = buttonSecondaryActive;
var buttonSuccessHover = 'var(--plasma-colors-button-success-hover)';
exports.buttonSuccessHover = buttonSuccessHover;
var buttonSuccessActive = 'var(--plasma-colors-button-success-active)';
exports.buttonSuccessActive = buttonSuccessActive;
var buttonWarningHover = 'var(--plasma-colors-button-warning-hover)';
exports.buttonWarningHover = buttonWarningHover;
var buttonWarningActive = 'var(--plasma-colors-button-warning-active)';
exports.buttonWarningActive = buttonWarningActive;
var buttonCriticalHover = 'var(--plasma-colors-button-critical-hover)';
exports.buttonCriticalHover = buttonCriticalHover;
var buttonCriticalActive = 'var(--plasma-colors-button-critical-active)';
exports.buttonCriticalActive = buttonCriticalActive;
var buttonCheckedHover = 'var(--plasma-colors-button-checked-hover)';
exports.buttonCheckedHover = buttonCheckedHover;
var buttonCheckedHoverColor = 'var(--plasma-colors-button-checked-hover-color)';
exports.buttonCheckedHoverColor = buttonCheckedHoverColor;
var buttonCheckedActive = 'var(--plasma-colors-button-checked-active)';
exports.buttonCheckedActive = buttonCheckedActive;
var buttonCheckedActiveColor = 'var(--plasma-colors-button-checked-active-color)';
exports.buttonCheckedActiveColor = buttonCheckedActiveColor;
var fieldColor = 'var(--plasma-field-color)';
exports.fieldColor = fieldColor;
var fieldLabelColor = 'var(--plasma-field-label-color)';
exports.fieldLabelColor = fieldLabelColor;
var inputBackground = 'var(--plasma-input-background-color)';
exports.inputBackground = inputBackground;
var inputBackgroundHover = 'var(--plasma-input-hover-background-color)';
exports.inputBackgroundHover = inputBackgroundHover;
var inputBackgroundFocus = 'var(--plasma-input-focus-background-color)';
exports.inputBackgroundFocus = inputBackgroundFocus;
var inputColor = 'var(--plasma-input-color)';
exports.inputColor = inputColor;
var inputCaretColor = 'var(--plasma-input-caret-color)';
exports.inputCaretColor = inputCaretColor;
var inputPlaceholderColor = 'var(--plasma-input-placeholder-color)';
exports.inputPlaceholderColor = inputPlaceholderColor;
var inputFocusPlaceholderColor = 'var(--plasma-input-focus-placeholder-color)';
exports.inputFocusPlaceholderColor = inputFocusPlaceholderColor;
var inputBorder = 'var(--plasma-colors-input-border)';
exports.inputBorder = inputBorder;
var inputBorderHover = 'var(--plasma-colors-input-border-hover)';
exports.inputBorderHover = inputBorderHover;
var inputBorderFocus = 'var(--plasma-colors-input-border-focus)';
exports.inputBorderFocus = inputBorderFocus;
var inputSuccessBackground = 'var(--plasma-input-success-background)';
exports.inputSuccessBackground = inputSuccessBackground;
var inputSuccessColor = 'var(--plasma-input-success-color)';
exports.inputSuccessColor = inputSuccessColor;
var inputSuccessCaretColor = 'var(--plasma-input-success-caret-color)';
exports.inputSuccessCaretColor = inputSuccessCaretColor;
var inputSuccessBorderColor = 'var(--plasma-input-success-border-color)';
exports.inputSuccessBorderColor = inputSuccessBorderColor;
var inputWarningBackground = 'var(--plasma-input-warning-background)';
exports.inputWarningBackground = inputWarningBackground;
var inputWarningColor = 'var(--plasma-input-warning-color)';
exports.inputWarningColor = inputWarningColor;
var inputWarningCaretColor = 'var(--plasma-input-warning-caret-color)';
exports.inputWarningCaretColor = inputWarningCaretColor;
var inputWarningBorderColor = 'var(--plasma-input-warning-border-color)';
exports.inputWarningBorderColor = inputWarningBorderColor;
var inputErrorBackground = 'var(--plasma-input-error-background)';
exports.inputErrorBackground = inputErrorBackground;
var inputErrorColor = 'var(--plasma-input-error-color)';
exports.inputErrorColor = inputErrorColor;
var inputErrorCaretColor = 'var(--plasma-input-error-caret-color)';
exports.inputErrorCaretColor = inputErrorCaretColor;
var inputErrorBorderColor = 'var(--plasma-input-error-border-color)';
exports.inputErrorBorderColor = inputErrorBorderColor;
var inputLabelColor = 'var(--plasma-input-label-color)';
exports.inputLabelColor = inputLabelColor;
var link = 'var(--plasma-colors-link)';
exports.link = link;
var linkHover = 'var(--plasma-colors-link-hover)';
exports.linkHover = linkHover;
var linkActive = 'var(--plasma-colors-link-active)';
exports.linkActive = linkActive;
var linkVisited = 'var(--plasma-colors-link-visited)';
exports.linkVisited = linkVisited;
var linkVisitedHover = 'var(--plasma-colors-link-visited-hover)';
exports.linkVisitedHover = linkVisitedHover;
var linkVisitedActive = 'var(--plasma-colors-link-visited-active)';
exports.linkVisitedActive = linkVisitedActive;
