// @sberdevices/plasma-ui/components/Pickers/TimePicker.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TimePicker = void 0;

var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));

var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));

var _plasmaCore = /*#__PURE__*/require("@sberdevices/plasma-core");

var _PickerDots = /*#__PURE__*/require("./PickerDots");

var _SimpleTimePicker = /*#__PURE__*/require("./SimpleTimePicker");

var _utils = /*#__PURE__*/require("./utils");

var _excluded = ["id", "options", "step", "size", "value", "min", "max", "disabled", "controls", "autofocus", "scrollSnapType", "visibleItems", "onChange", "name", "enableNativeControl", "secondsAriaLabel", "minutesAriaLabel", "hoursAriaLabel", "infiniteScroll"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }

function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }

function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

var StyledWrapper = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-7uw1zz-0"
})(["display:flex;width:max-content;align-items:stretch;"]);

var defaultOptions = {
  hours: true,
  minutes: true,
  seconds: true
};
/**
 * Вернет массив чисел от `from` до `to` с интервалом `step`.
 */

var getRange = function getRange(from, to, step) {
  var range = [];

  for (var i = from; i <= to; i += step) {
    range.push(i);
  }

  return range;
};
/**
 * Сравнит число с массивом чисел и вернет значение массива,
 * максимальное близкое заданному числу.
 */


var getClosestValue = function getClosestValue(range, value) {
  if (value === 0) {
    return range[0];
  }

  var weights = range.map(function (i) {
    return value <= i ? value / i : i / value;
  });
  return range[weights.indexOf(Math.max.apply(Math, _toConsumableArray(weights)))];
};
/**
 * Вернёт секунды
 */


var getSeconds = function getSeconds(_ref) {
  var _ref2 = _slicedToArray(_ref, 3),
      hours = _ref2[0],
      minutes = _ref2[1],
      seconds = _ref2[2];

  return hours * 60 * 60 + minutes * 60 + seconds;
};
/**
 * Для того, чтобы значение не выпадало из диапозона,
 * надо выставить в соответствии с последним
 */


var getValuesInRange = function getValuesInRange(_ref3, _ref4, value) {
  var _ref5 = _slicedToArray(_ref3, 3),
      hoursRange = _ref5[0],
      minsRange = _ref5[1],
      secsRange = _ref5[2];

  var _ref6 = _slicedToArray(_ref4, 3),
      hours = _ref6[0],
      minutes = _ref6[1],
      seconds = _ref6[2];

  if (hoursRange.indexOf(hours) === -1 || minsRange.indexOf(minutes) === -1 || secsRange.indexOf(seconds) === -1) {
    var newHours = hoursRange.indexOf(hours) === -1 ? getClosestValue(hoursRange, hours) : hours;
    var newMins = minsRange.indexOf(minutes) === -1 ? getClosestValue(minsRange, minutes) : minutes;
    var newSecs = secsRange.indexOf(seconds) === -1 ? getClosestValue(secsRange, seconds) : seconds; // eslint-disable-next-line no-restricted-globals

    if (isNaN(newHours) || isNaN(newMins) || isNaN(newSecs)) {
      throw new Error("Passed value ".concat(value, " is out of range"));
    }

    return [newHours, newMins, newSecs];
  }

  return [hours, minutes, seconds];
};

/**
 * Компонент для выбора времени.
 */
var TimePicker = function TimePicker(_ref7) {
  var id = _ref7.id,
      _ref7$options = _ref7.options,
      options = _ref7$options === void 0 ? defaultOptions : _ref7$options,
      step = _ref7.step,
      size = _ref7.size,
      value = _ref7.value,
      min = _ref7.min,
      max = _ref7.max,
      disabled = _ref7.disabled,
      controls = _ref7.controls,
      autofocus = _ref7.autofocus,
      scrollSnapType = _ref7.scrollSnapType,
      visibleItems = _ref7.visibleItems,
      onChange = _ref7.onChange,
      name = _ref7.name,
      enableNativeControl = _ref7.enableNativeControl,
      secondsAriaLabel = _ref7.secondsAriaLabel,
      minutesAriaLabel = _ref7.minutesAriaLabel,
      hoursAriaLabel = _ref7.hoursAriaLabel,
      infiniteScroll = _ref7.infiniteScroll,
      rest = _objectWithoutProperties(_ref7, _excluded);

  var normalizeValues = _react["default"].useMemo(function () {
    return (0, _utils.getNormalizeValues)(_utils.getTimeValues, getSeconds)(value, min, max);
  }, [value, min, max]);

  var _React$useState = _react["default"].useState(normalizeValues),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      _React$useState2$ = _slicedToArray(_React$useState2[0], 3),
      hours = _React$useState2$[0],
      minutes = _React$useState2$[1],
      seconds = _React$useState2$[2],
      setState = _React$useState2[1];

  var _getTimeValues = (0, _utils.getTimeValues)(min),
      _getTimeValues2 = _slicedToArray(_getTimeValues, 3),
      minHours = _getTimeValues2[0],
      minMinutes = _getTimeValues2[1],
      minSeconds = _getTimeValues2[2];

  var _getTimeValues3 = (0, _utils.getTimeValues)(max),
      _getTimeValues4 = _slicedToArray(_getTimeValues3, 3),
      maxHours = _getTimeValues4[0],
      maxMinutes = _getTimeValues4[1],
      maxSeconds = _getTimeValues4[2]; // Диапазоны для списков зависят от min и max,
  // при чем min и max принимаются как возможные предельные значения,
  // а не как контейнеры для компонент hours, minutes, seconds


  var _React$useMemo = _react["default"].useMemo(function () {
    var minMins = 0;
    var maxMins = 59;
    var minSecs = 0;
    var maxSecs = 59;

    if (hours === minHours) {
      minMins = minMinutes;
    }

    if (hours === maxHours) {
      maxMins = maxMinutes;
    }

    if (hours === minHours && minutes === minMinutes) {
      minSecs = minSeconds;
    }

    if (hours === maxHours && minutes === maxMinutes) {
      maxSecs = maxSeconds;
    }

    var hoursStep = 1;
    var minsStep = 1;
    var secsStep = 1;

    if (step) {
      var hoursMod = step % 3600;
      var minsMod = hoursMod % 60;
      hoursStep = (step - hoursMod) / 3600 || 1;
      minsStep = (hoursMod - minsMod) / 60 || 1;
      secsStep = minsMod || 1;
    }

    return [getRange(minHours, maxHours, hoursStep), getRange(minMins, maxMins, minsStep), getRange(minSecs, maxSecs, secsStep)];
  }, [minHours, maxHours, minMinutes, maxMinutes, minSeconds, maxSeconds, hours, minutes, step]),
      _React$useMemo2 = _slicedToArray(_React$useMemo, 3),
      hoursRange = _React$useMemo2[0],
      minsRange = _React$useMemo2[1],
      secsRange = _React$useMemo2[2];

  var onHoursChange = _react["default"].useCallback(function (_ref8) {
    var h = _ref8.value;
    return setState(function (_ref9) {
      var _ref10 = _slicedToArray(_ref9, 3),
          m = _ref10[1],
          s = _ref10[2];

      return [h, m, s];
    });
  }, []);

  var onMinutesChange = _react["default"].useCallback(function (_ref11) {
    var m = _ref11.value;
    return setState(function (_ref12) {
      var _ref13 = _slicedToArray(_ref12, 3),
          h = _ref13[0],
          s = _ref13[2];

      return [h, m, s];
    });
  }, []);

  var onSecondsChange = _react["default"].useCallback(function (_ref14) {
    var s = _ref14.value;
    return setState(function (_ref15) {
      var _ref16 = _slicedToArray(_ref15, 2),
          h = _ref16[0],
          m = _ref16[1];

      return [h, m, s];
    });
  }, []); // При очередном прогоне, если значения hours, minutes, seconds изменились,
  // необходимо вызвать событие изменения, создав новый экземпляр Date


  (0, _plasmaCore.useIsomorphicLayoutEffect)(function () {
    var oldTime = normalizeValues;

    if (onChange && (0, _utils.isChanged)(oldTime, [hours, minutes, seconds])) {
      var newValue = new Date(value);
      newValue.setHours(hours);
      newValue.setMinutes(minutes);
      newValue.setSeconds(seconds);
      onChange(newValue);
    }
  }, [hours, minutes, seconds]);
  /**
   * Если значение value обновилось извне, необходимо изменить стейт
   * и вызвать событие изменения, создав новый экземпляр Date
   */

  (0, _plasmaCore.useIsomorphicLayoutEffect)(function () {
    setState(function (prevTime) {
      var _normalizeValues = _slicedToArray(normalizeValues, 3),
          newHours = _normalizeValues[0],
          newMins = _normalizeValues[1],
          newSecs = _normalizeValues[2];

      if (!(0, _utils.isChanged)(prevTime, [newHours, newMins, newSecs])) {
        return prevTime;
      }

      if (onChange) {
        var newValue = new Date(value);
        newValue.setHours(newHours);
        newValue.setMinutes(newMins);
        newValue.setSeconds(newSecs);
        onChange(newValue);
      }

      return [newHours, newMins, newSecs];
    });
  }, [value, normalizeValues, min, max]);
  var newTime = getValuesInRange([hoursRange, minsRange, secsRange], [hours, minutes, seconds], value);

  if ((0, _utils.isChanged)([hours, minutes, seconds], newTime)) {
    setState(newTime);
  }

  return /*#__PURE__*/_react["default"].createElement(StyledWrapper, _extends({
    id: id
  }, rest), options.hours && /*#__PURE__*/_react["default"].createElement(_SimpleTimePicker.SimpleTimePicker, {
    id: id,
    type: "hours",
    autofocus: autofocus,
    disabled: disabled,
    controls: controls,
    visibleItems: visibleItems,
    size: size,
    range: hoursRange,
    value: hours,
    scrollSnapType: scrollSnapType,
    infiniteScroll: infiniteScroll,
    onChange: onHoursChange,
    "aria-label": hoursAriaLabel
  }), options.hours && options.minutes && /*#__PURE__*/_react["default"].createElement(_PickerDots.PickerDots, {
    $size: size
  }), options.minutes && /*#__PURE__*/_react["default"].createElement(_SimpleTimePicker.SimpleTimePicker, {
    id: id,
    type: "minutes",
    autofocus: autofocus && !options.hours,
    disabled: disabled,
    controls: controls,
    visibleItems: visibleItems,
    size: size,
    range: minsRange,
    value: minutes,
    scrollSnapType: scrollSnapType,
    infiniteScroll: infiniteScroll,
    onChange: onMinutesChange,
    "aria-label": minutesAriaLabel
  }), options.minutes && options.seconds && /*#__PURE__*/_react["default"].createElement(_PickerDots.PickerDots, {
    $size: size
  }), options.seconds && /*#__PURE__*/_react["default"].createElement(_SimpleTimePicker.SimpleTimePicker, {
    id: id,
    type: "seconds",
    autofocus: autofocus && !options.hours && !options.minutes,
    disabled: disabled,
    controls: controls,
    visibleItems: visibleItems,
    size: size,
    range: secsRange,
    value: seconds,
    scrollSnapType: scrollSnapType,
    infiniteScroll: infiniteScroll,
    onChange: onSecondsChange,
    "aria-label": secondsAriaLabel
  }), enableNativeControl && /*#__PURE__*/_react["default"].createElement("input", {
    type: "hidden",
    value: value.toISOString(),
    name: name
  }));
};

exports.TimePicker = TimePicker;
