// @sberdevices/plasma-web/components/Typography/Headline/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Headline1", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.Headline1;
  }
});
Object.defineProperty(exports, "Headline2", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.Headline2;
  }
});
Object.defineProperty(exports, "Headline3", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.Headline3;
  }
});
Object.defineProperty(exports, "Headline4", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.Headline4;
  }
});
exports.H5 = exports.Headline5 = void 0;

var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));

var _tokens = /*#__PURE__*/require("../../../tokens");

var _plasmaCore = /*#__PURE__*/require("@sberdevices/plasma-core");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var Headline5 = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web__sc-euyaij-0"
})( /*#__PURE__*/_objectSpread({
  overflowWrap: 'break-word',
  hyphens: 'auto'
}, _tokens.headline5));

exports.Headline5 = Headline5;

var H5 = /*#__PURE__*/_styledComponents["default"].h3.withConfig({
  componentId: "plasma-web__sc-euyaij-1"
})( /*#__PURE__*/_objectSpread({
  margin: 0
}, _tokens.headline5));

exports.H5 = H5;
