// @sberdevices/plasma-ui/utils/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "animatedScrollToX", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.animatedScrollToX;
  }
});
Object.defineProperty(exports, "animatedScrollToY", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.animatedScrollToY;
  }
});
Object.defineProperty(exports, "convertRoundnessMatrix", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.convertRoundnessMatrix;
  }
});
Object.defineProperty(exports, "padZeroNumber", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.padZeroNumber;
  }
});
Object.defineProperty(exports, "monthLongName", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.monthLongName;
  }
});
Object.defineProperty(exports, "formatCurrency", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.formatCurrency;
  }
});
Object.defineProperty(exports, "mediaQuery", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.mediaQuery;
  }
});
Object.defineProperty(exports, "gridBreakpoints", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.gridBreakpoints;
  }
});
Object.defineProperty(exports, "gridColumns", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.gridColumns;
  }
});
Object.defineProperty(exports, "gridMargins", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.gridMargins;
  }
});
Object.defineProperty(exports, "gridGutters", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.gridGutters;
  }
});
Object.defineProperty(exports, "gridSizes", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.gridSizes;
  }
});
Object.defineProperty(exports, "setRef", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.setRef;
  }
});
Object.defineProperty(exports, "toCssSize", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.toCssSize;
  }
});
Object.defineProperty(exports, "transformStyles", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.transformStyles;
  }
});
Object.defineProperty(exports, "detectDevice", {
  enumerable: true,
  get: function get() {
    return _deviceDetection.detectDevice;
  }
});
Object.defineProperty(exports, "deviceScales", {
  enumerable: true,
  get: function get() {
    return _deviceDetection.deviceScales;
  }
});
Object.defineProperty(exports, "isSberBox", {
  enumerable: true,
  get: function get() {
    return _deviceDetection.isSberBox;
  }
});
Object.defineProperty(exports, "isSberPortal", {
  enumerable: true,
  get: function get() {
    return _deviceDetection.isSberPortal;
  }
});

var _plasmaCore = /*#__PURE__*/require("@sberdevices/plasma-core");

var _deviceDetection = /*#__PURE__*/require("./deviceDetection");
