// @sberdevices/plasma-web/components/Checkbox/Checkbox.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Checkbox = exports.StyledDescription = exports.StyledLabel = exports.StyledContent = exports.StyledTrigger = exports.StyledInput = exports.StyledRoot = exports.syntheticFocus = void 0;

var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));

var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));

var _plasmaCore = /*#__PURE__*/require("@sberdevices/plasma-core");

var _utils = /*#__PURE__*/require("../../utils");

var _Icons = /*#__PURE__*/require("./Icons");

var _excluded = ["id", "label", "description", "indeterminate", "disabled", "style", "className", "aria-label"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var syntheticFocus = function syntheticFocus(ruleset, focused) {
  return (0, _styledComponents.css)(["input[data-focus-visible-added] + label &{outline:none;", "}", ";"], ruleset, focused && ruleset);
};

exports.syntheticFocus = syntheticFocus;
var StyledRoot = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaCore.BaseboxRoot).withConfig({
  componentId: "plasma-web__sc-1x2r0l2-0"
})(["margin-left:0.1875rem;margin-bottom:0.1875rem;", ""], function (_ref) {
  var $disabled = _ref.$disabled;
  return $disabled && 'cursor: not-allowed;';
});
exports.StyledRoot = StyledRoot;
var StyledInput = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaCore.BaseboxInput).withConfig({
  componentId: "plasma-web__sc-1x2r0l2-1"
})(["width:1px;height:1px;overflow:hidden;clip:rect(0 0 0 0);"]);
exports.StyledInput = StyledInput;
var StyledTrigger = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaCore.BaseboxTrigger).withConfig({
  componentId: "plasma-web__sc-1x2r0l2-2"
})(["margin:0.1875rem 0;width:1.125rem;height:1.125rem;background:", ";border:0.125rem solid ", ";border-radius:0.25rem;color:", ";flex-shrink:0;flex-grow:0;input[type='checkbox']:indeterminate + label &{background:", ";border-color:", ";color:", ";}input:disabled + label &{background:", ";border-color:", ";color:", ";cursor:inherit;}input:checked + label &{background:", ";border-color:", ";color:", ";}input:checked:disabled + label &{background:", ";border-color:", ";color:", ";}", ""], _plasmaCore.transparent, _plasmaCore.tertiary, _plasmaCore.tertiary, _plasmaCore.transparent, _plasmaCore.accent, _plasmaCore.accent, _plasmaCore.buttonSecondary, _plasmaCore.transparent, _plasmaCore.transparent, _plasmaCore.accent, _plasmaCore.accent, _plasmaCore.white, _plasmaCore.buttonSecondary, _plasmaCore.transparent, _plasmaCore.tertiary, function (_ref2) {
  var _ref2$outlineRadius = _ref2.outlineRadius,
      outlineRadius = _ref2$outlineRadius === void 0 ? '0.25rem' : _ref2$outlineRadius;
  return (0, _styledComponents.css)(["", ""], (0, _plasmaCore.addFocus)({
    synthesizeFocus: syntheticFocus,
    outlineOffset: '0.25rem',
    outlineRadius: outlineRadius
  }));
});
exports.StyledTrigger = StyledTrigger;
var StyledContent = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaCore.BaseboxContent).withConfig({
  componentId: "plasma-web__sc-1x2r0l2-3"
})(["margin-left:0.875rem;input:disabled + label &{opacity:0.4;}"]);
exports.StyledContent = StyledContent;
var StyledLabel = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaCore.BaseboxLabel).withConfig({
  componentId: "plasma-web__sc-1x2r0l2-4"
})(["line-height:1.5rem;"]);
exports.StyledLabel = StyledLabel;
var StyledDescription = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaCore.BaseboxDescription).withConfig({
  componentId: "plasma-web__sc-1x2r0l2-5"
})(["input:disabled + label &{opacity:0.4;}"]);
exports.StyledDescription = StyledDescription;
var StyledDone = /*#__PURE__*/(0, _styledComponents["default"])(_Icons.Done).withConfig({
  componentId: "plasma-web__sc-1x2r0l2-6"
})(["position:absolute;top:-0.125rem;left:-0.125rem;transform:scale(0);transition:transform 0.15s ease-in-out;input:checked + label ", " &{transform:scale(1);}"], _plasmaCore.BaseboxTrigger);
var StyledIndeterminate = /*#__PURE__*/(0, _styledComponents["default"])(_Icons.Indeterminate).withConfig({
  componentId: "plasma-web__sc-1x2r0l2-7"
})(["position:absolute;top:-0.125rem;left:-0.125rem;"]);
/**
 * Флажок или *чекбокс*. Позволяет пользователю управлять параметром с двумя состояниями — ☑ включено и ☐ отключено.
 */
// eslint-disable-next-line prefer-arrow-callback

var Checkbox = /*#__PURE__*/(0, _react.forwardRef)(function Checkbox(_ref3, ref) {
  var id = _ref3.id,
      label = _ref3.label,
      description = _ref3.description,
      indeterminate = _ref3.indeterminate,
      disabled = _ref3.disabled,
      style = _ref3.style,
      className = _ref3.className,
      ariaLabelExternal = _ref3['aria-label'],
      rest = _objectWithoutProperties(_ref3, _excluded);

  var inputRef = (0, _react.useRef)(null);
  var forkRef = (0, _plasmaCore.useForkRef)(inputRef, ref);

  _react["default"].useEffect(function () {
    if (inputRef.current) {
      inputRef.current.indeterminate = Boolean(indeterminate);
    }
  }, [inputRef, indeterminate]);

  var uniqId = (0, _plasmaCore.useUniqId)();
  var uniqLabelId = (0, _plasmaCore.useUniqId)();
  var uniqDescriptionId = (0, _plasmaCore.useUniqId)();
  var checkboxId = id || uniqId;
  var ariaLabel = (0, _react.useMemo)(function () {
    return ariaLabelExternal || (0, _utils.extractTextFrom)(label);
  }, [ariaLabelExternal, label]);
  return /*#__PURE__*/_react["default"].createElement(StyledRoot, {
    $disabled: disabled,
    style: style,
    className: className,
    tabIndex: -1
  }, /*#__PURE__*/_react["default"].createElement(StyledInput, _extends({}, rest, {
    id: checkboxId,
    ref: forkRef,
    type: "checkbox",
    disabled: disabled,
    "aria-label": ariaLabel,
    "aria-describedby": uniqDescriptionId
  })), /*#__PURE__*/_react["default"].createElement(_plasmaCore.BaseboxContentWrapper, {
    htmlFor: checkboxId
  }, /*#__PURE__*/_react["default"].createElement(StyledTrigger, null, indeterminate ? /*#__PURE__*/_react["default"].createElement(StyledIndeterminate, null) : /*#__PURE__*/_react["default"].createElement(StyledDone, null)), label && /*#__PURE__*/_react["default"].createElement(StyledContent, null, /*#__PURE__*/_react["default"].createElement(StyledLabel, {
    as: "span",
    id: uniqLabelId,
    "aria-hidden": typeof label === 'string'
  }, label), description && /*#__PURE__*/_react["default"].createElement(_plasmaCore.BaseboxDescription, {
    mt: 4,
    id: uniqDescriptionId
  }, description))));
});
exports.Checkbox = Checkbox;
