// @sberdevices/plasma-ui/es/components/ProductCard/ProductCard.js
var _excluded = ["badge", "media", "text", "price", "oldPrice", "quantity", "quantityStep", "quantityMin", "quantityMax", "onQuantityChange", "disabled", "backgroundColor"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { forwardRef } from 'react';
import styled, { css } from 'styled-components';
import { body1, black, blackSecondary, success, mediaQuery } from '@sberdevices/plasma-core';
import Color from 'color';
import { Price } from '../Price';
import { Footnote1, Body2, Caption } from '../Typography';
import { Card } from '../Card/Card';
import { CardBody } from '../Card/CardBody';
import { CardContent } from '../Card/CardContent';
import { ProductCardStepper } from './ProductCardStepper';
var StyledCard = /*#__PURE__*/styled(Card).withConfig({
  componentId: "plasma-ui__sc-lks0gg-0"
})(["background-color:", ";color:", ";"], function (_ref) {
  var $backgroundColor = _ref.$backgroundColor;
  return $backgroundColor;
}, blackSecondary);
var StyledMediaSlot = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-ui__sc-lks0gg-1"
})(["& img{display:block;}"]);

var getGradient = function getGradient(backgroundColor) {
  var color = Color(backgroundColor);
  return "linear-gradient(\n        180deg,\n        ".concat(color.alpha(0).string(), " 0%,\n        ").concat(color.alpha(0.0156863).string(), " 8.62%,\n        ").concat(color.alpha(0.054902).string(), " 16.56%,\n        ").concat(color.alpha(0.117647).string(), " 23.93%,\n        ").concat(color.alpha(0.2).string(), " 30.85%,\n        ").concat(color.alpha(0.290196).string(), " 37.42%,\n        ").concat(color.alpha(0.392157).string(), " 43.77%,\n        ").concat(color.alpha(0.501961).string(), " 50%,\n        ").concat(color.alpha(0.607843).string(), " 56.23%,\n        ").concat(color.alpha(0.709804).string(), " 62.58%,\n        ").concat(color.alpha(0.8).string(), " 69.15%,\n        ").concat(color.alpha(0.882353).string(), " 76.07%,\n        ").concat(color.alpha(0.945098).string(), " 83.44%,\n        ").concat(color.alpha(0.984314).string(), " 91.38%,\n        ").concat(backgroundColor, " 100%\n    )");
};

var StyledCardContent = /*#__PURE__*/styled(CardContent).withConfig({
  componentId: "plasma-ui__sc-lks0gg-2"
})(["padding:0.75rem;border-radius:0;background-color:", ";transition:", ";&::after{content:'';position:absolute;top:-2.5rem;left:0;right:0;height:2.5rem;opacity:0;transition:", ";background:", ";}", ""], function (_ref2) {
  var $backgroundColor = _ref2.$backgroundColor;
  return $backgroundColor;
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.lowPerformance ? 'unset' : 'all 0.15s ease-in-out';
}, function (_ref4) {
  var theme = _ref4.theme;
  return theme.lowPerformance ? 'unset' : 'all 0.15s ease-in-out';
}, function (_ref5) {
  var $backgroundColor = _ref5.$backgroundColor;
  return $backgroundColor && getGradient($backgroundColor);
}, function (_ref6) {
  var $cover = _ref6.$cover;
  return $cover && css(["margin-top:-1.875rem;&::after{opacity:1;}"]);
});
var StyledBadgeSlot = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-ui__sc-lks0gg-3"
})(["position:absolute;top:0.5rem;left:0.5rem;"]);
var StyledText = /*#__PURE__*/styled(Footnote1).withConfig({
  componentId: "plasma-ui__sc-lks0gg-4"
})(["", ""], function (props) {
  return mediaQuery('XL', props.theme.deviceScale)(css(["", ""], body1));
});
var StyledBottom = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-ui__sc-lks0gg-5"
})(["display:flex;align-items:center;flex-direction:column;margin-top:0.25rem;"]);
var StyledPrices = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-ui__sc-lks0gg-6"
})(["display:flex;flex-direction:column;justify-content:center;width:100%;min-height:2.375rem;"]);
var StyledPrice = /*#__PURE__*/styled(Price).withConfig({
  componentId: "plasma-ui__sc-lks0gg-7"
})(["color:", ";"], function (_ref7) {
  var $type = _ref7.$type;

  switch ($type) {
    case 'new':
      return success;

    case 'old':
      return blackSecondary;

    default:
      return black;
  }
});
var StyledStepper = /*#__PURE__*/styled(ProductCardStepper).withConfig({
  componentId: "plasma-ui__sc-lks0gg-8"
})(["width:100%;transition:", ";", ""], function (_ref8) {
  var theme = _ref8.theme;
  return theme.lowPerformance ? 'unset' : 'all 0.15s ease-in-out';
}, function (_ref9) {
  var $onTop = _ref9.$onTop;
  return $onTop ? css(["margin-top:-2.5rem;"]) : css(["margin-top:0.5rem;"]);
});
/**
 * Карточка продукта с возможностью указания картинки, текста, цены и выбора количества.
 */
// eslint-disable-next-line prefer-arrow-callback

export var ProductCard = /*#__PURE__*/forwardRef(function ProductCard(_ref10, ref) {
  var badge = _ref10.badge,
      media = _ref10.media,
      text = _ref10.text,
      price = _ref10.price,
      oldPrice = _ref10.oldPrice,
      quantity = _ref10.quantity,
      quantityStep = _ref10.quantityStep,
      quantityMin = _ref10.quantityMin,
      quantityMax = _ref10.quantityMax,
      onQuantityChange = _ref10.onQuantityChange,
      disabled = _ref10.disabled,
      _ref10$backgroundColo = _ref10.backgroundColor,
      backgroundColor = _ref10$backgroundColo === void 0 ? '#FFFFFF' : _ref10$backgroundColo,
      rest = _objectWithoutProperties(_ref10, _excluded);

  return /*#__PURE__*/React.createElement(StyledCard, _extends({}, rest, {
    ref: ref,
    disabled: disabled,
    $backgroundColor: backgroundColor
  }), /*#__PURE__*/React.createElement(CardBody, null, media && /*#__PURE__*/React.createElement(StyledMediaSlot, null, media), badge && /*#__PURE__*/React.createElement(StyledBadgeSlot, null, badge), /*#__PURE__*/React.createElement(StyledCardContent, {
    $backgroundColor: backgroundColor,
    $cover: media !== undefined && quantity !== undefined && quantity > 0
  }, text && /*#__PURE__*/React.createElement(StyledText, null, text), (price !== undefined || quantity !== undefined) && /*#__PURE__*/React.createElement(StyledBottom, null, price !== undefined && /*#__PURE__*/React.createElement(StyledPrices, null, /*#__PURE__*/React.createElement(StyledPrice, {
    $type: oldPrice !== undefined ? 'new' : undefined,
    forwardedAs: Body2
  }, price), oldPrice && /*#__PURE__*/React.createElement(StyledPrice, {
    $type: "old",
    stroke: true,
    forwardedAs: Caption
  }, oldPrice)), quantity !== undefined && /*#__PURE__*/React.createElement(StyledStepper, {
    value: quantity,
    step: quantityStep,
    min: quantityMin,
    max: quantityMax,
    onChange: onQuantityChange,
    $onTop: price !== undefined && quantity === 0,
    disabled: disabled
  })))));
});
