// @sberdevices/plasma-tokens/themes/index.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
var darkSber_1 = require("./darkSber");
Object.defineProperty(exports, "darkSber", { enumerable: true, get: function () { return darkSber_1.darkSber; } });
var darkEva_1 = require("./darkEva");
Object.defineProperty(exports, "darkEva", { enumerable: true, get: function () { return darkEva_1.darkEva; } });
var darkJoy_1 = require("./darkJoy");
Object.defineProperty(exports, "darkJoy", { enumerable: true, get: function () { return darkJoy_1.darkJoy; } });
var darkBrand_1 = require("./darkBrand");
Object.defineProperty(exports, "darkBrand", { enumerable: true, get: function () { return darkBrand_1.darkBrand; } });
var lightSber_1 = require("./lightSber");
Object.defineProperty(exports, "lightSber", { enumerable: true, get: function () { return lightSber_1.lightSber; } });
var lightEva_1 = require("./lightEva");
Object.defineProperty(exports, "lightEva", { enumerable: true, get: function () { return lightEva_1.lightEva; } });
var lightJoy_1 = require("./lightJoy");
Object.defineProperty(exports, "lightJoy", { enumerable: true, get: function () { return lightJoy_1.lightJoy; } });
var lightBrand_1 = require("./lightBrand");
Object.defineProperty(exports, "lightBrand", { enumerable: true, get: function () { return lightBrand_1.lightBrand; } });
