// @sberdevices/plasma-web/es/components/Radiobox/Radiobox.js
var _excluded = ["id", "label", "description", "disabled", "style", "className", "aria-label"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { forwardRef, useMemo } from 'react';
import styled from 'styled-components';
import { BaseboxDescription, BaseboxContentWrapper, useUniqId, white } from '@sberdevices/plasma-core';
import { extractTextFrom } from '../../utils';
import { StyledRoot as CheckboxRoot, StyledInput as CheckboxInput, StyledTrigger as CheckboxTrigger, StyledContent as CheckboxContent, StyledLabel as CheckboxLabel } from '../Checkbox/Checkbox';
var StyledTrigger = /*#__PURE__*/styled(CheckboxTrigger).attrs({
  outlineRadius: '1.125rem'
}).withConfig({
  componentId: "plasma-web__sc-2033cg-0"
})(["flex-shrink:0;width:1.25rem;height:1.25rem;border-radius:1.125rem;"]);
var StyledLabel = /*#__PURE__*/styled(CheckboxLabel).withConfig({
  componentId: "plasma-web__sc-2033cg-1"
})(["line-height:1.625rem;"]);
var StyledEllipse = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-web__sc-2033cg-2"
})(["position:absolute;top:0;left:0;right:0;bottom:0;margin:auto;width:0.625rem;height:0.625rem;border-radius:0.625rem;background-color:", ";transition:transform 0.3s ease-in-out;transform:scale(0);input:checked + label ", " &{transform:scale(1);}"], white, StyledTrigger);
/**
 * Переключатель, или *радиокнопка*.
 */
// eslint-disable-next-line prefer-arrow-callback

export var Radiobox = /*#__PURE__*/forwardRef(function Radiobox(_ref, ref) {
  var id = _ref.id,
      label = _ref.label,
      description = _ref.description,
      disabled = _ref.disabled,
      style = _ref.style,
      className = _ref.className,
      ariaLabelExternal = _ref['aria-label'],
      rest = _objectWithoutProperties(_ref, _excluded);

  var uniqId = useUniqId();
  var uniqLabelId = useUniqId();
  var uniqDescriptionId = useUniqId();
  var radioboxId = id || uniqId;
  var ariaLabel = useMemo(function () {
    return ariaLabelExternal || extractTextFrom(label);
  }, [ariaLabelExternal, label]);
  return /*#__PURE__*/React.createElement(CheckboxRoot, {
    $disabled: disabled,
    style: style,
    className: className,
    tabIndex: -1
  }, /*#__PURE__*/React.createElement(CheckboxInput, _extends({}, rest, {
    id: radioboxId,
    ref: ref,
    type: "radio",
    disabled: disabled,
    "aria-label": ariaLabel,
    "aria-describedby": uniqDescriptionId
  })), /*#__PURE__*/React.createElement(BaseboxContentWrapper, {
    htmlFor: radioboxId
  }, /*#__PURE__*/React.createElement(StyledTrigger, null, /*#__PURE__*/React.createElement(StyledEllipse, null)), label && /*#__PURE__*/React.createElement(CheckboxContent, null, /*#__PURE__*/React.createElement(StyledLabel, {
    as: "span",
    id: uniqLabelId,
    "aria-hidden": typeof label === 'string'
  }, label), description && /*#__PURE__*/React.createElement(BaseboxDescription, {
    mt: 4,
    id: uniqDescriptionId
  }, description))));
});
