// @sberdevices/plasma-ui/components/Carousel/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "CarouselGridWrapper", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.CarouselGridWrapper;
  }
});
Object.defineProperty(exports, "CarouselItem", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.CarouselItem;
  }
});
Object.defineProperty(exports, "useCarouselItem", {
  enumerable: true,
  get: function get() {
    return _Carousel.useCarouselItem;
  }
});
Object.defineProperty(exports, "useCarouselContext", {
  enumerable: true,
  get: function get() {
    return _Carousel.useCarouselContext;
  }
});
Object.defineProperty(exports, "useRemoteHandlers", {
  enumerable: true,
  get: function get() {
    return _Carousel.useRemoteHandlers;
  }
});
Object.defineProperty(exports, "Carousel", {
  enumerable: true,
  get: function get() {
    return _Carousel2.Carousel;
  }
});
Object.defineProperty(exports, "CarouselCol", {
  enumerable: true,
  get: function get() {
    return _CarouselCol.CarouselCol;
  }
});

var _plasmaCore = /*#__PURE__*/require("@sberdevices/plasma-core");

var _Carousel = /*#__PURE__*/require("./Carousel.hooks");

var _Carousel2 = /*#__PURE__*/require("./Carousel");

var _CarouselCol = /*#__PURE__*/require("./CarouselCol");
