// @sberdevices/plasma-web/components/Radiobox/Radiobox.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Radiobox = void 0;

var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));

var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));

var _plasmaCore = /*#__PURE__*/require("@sberdevices/plasma-core");

var _utils = /*#__PURE__*/require("../../utils");

var _Checkbox = /*#__PURE__*/require("../Checkbox/Checkbox");

var _excluded = ["id", "label", "description", "disabled", "style", "className", "aria-label"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var StyledTrigger = /*#__PURE__*/(0, _styledComponents["default"])(_Checkbox.StyledTrigger).attrs({
  outlineRadius: '1.125rem'
}).withConfig({
  componentId: "plasma-web__sc-2033cg-0"
})(["flex-shrink:0;width:1.25rem;height:1.25rem;border-radius:1.125rem;"]);
var StyledLabel = /*#__PURE__*/(0, _styledComponents["default"])(_Checkbox.StyledLabel).withConfig({
  componentId: "plasma-web__sc-2033cg-1"
})(["line-height:1.625rem;"]);

var StyledEllipse = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web__sc-2033cg-2"
})(["position:absolute;top:0;left:0;right:0;bottom:0;margin:auto;width:0.625rem;height:0.625rem;border-radius:0.625rem;background-color:", ";transition:transform 0.3s ease-in-out;transform:scale(0);input:checked + label ", " &{transform:scale(1);}"], _plasmaCore.white, StyledTrigger);
/**
 * Переключатель, или *радиокнопка*.
 */
// eslint-disable-next-line prefer-arrow-callback


var Radiobox = /*#__PURE__*/(0, _react.forwardRef)(function Radiobox(_ref, ref) {
  var id = _ref.id,
      label = _ref.label,
      description = _ref.description,
      disabled = _ref.disabled,
      style = _ref.style,
      className = _ref.className,
      ariaLabelExternal = _ref['aria-label'],
      rest = _objectWithoutProperties(_ref, _excluded);

  var uniqId = (0, _plasmaCore.useUniqId)();
  var uniqLabelId = (0, _plasmaCore.useUniqId)();
  var uniqDescriptionId = (0, _plasmaCore.useUniqId)();
  var radioboxId = id || uniqId;
  var ariaLabel = (0, _react.useMemo)(function () {
    return ariaLabelExternal || (0, _utils.extractTextFrom)(label);
  }, [ariaLabelExternal, label]);
  return /*#__PURE__*/_react["default"].createElement(_Checkbox.StyledRoot, {
    $disabled: disabled,
    style: style,
    className: className,
    tabIndex: -1
  }, /*#__PURE__*/_react["default"].createElement(_Checkbox.StyledInput, _extends({}, rest, {
    id: radioboxId,
    ref: ref,
    type: "radio",
    disabled: disabled,
    "aria-label": ariaLabel,
    "aria-describedby": uniqDescriptionId
  })), /*#__PURE__*/_react["default"].createElement(_plasmaCore.BaseboxContentWrapper, {
    htmlFor: radioboxId
  }, /*#__PURE__*/_react["default"].createElement(StyledTrigger, null, /*#__PURE__*/_react["default"].createElement(StyledEllipse, null)), label && /*#__PURE__*/_react["default"].createElement(_Checkbox.StyledContent, null, /*#__PURE__*/_react["default"].createElement(StyledLabel, {
    as: "span",
    id: uniqLabelId,
    "aria-hidden": typeof label === 'string'
  }, label), description && /*#__PURE__*/_react["default"].createElement(_plasmaCore.BaseboxDescription, {
    mt: 4,
    id: uniqDescriptionId
  }, description))));
});
exports.Radiobox = Radiobox;
