// @sberdevices/plasma-web/components/Modal/Modal.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Modal = void 0;

var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));

var _reactDom = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react-dom"));

var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));

var _plasmaCore = /*#__PURE__*/require("@sberdevices/plasma-core");

var _ModalsContext = /*#__PURE__*/require("./ModalsContext");

var _ModalView = /*#__PURE__*/require("./ModalView");

var _excluded = ["id", "isOpen", "onClose"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var ESCAPE_KEYCODE = 27;
var wrapperHideAnimation = /*#__PURE__*/(0, _styledComponents.keyframes)(["from{opacity:1;}to{opacity:0;}"]);
var bodyShowAnimation = /*#__PURE__*/(0, _styledComponents.keyframes)(["from{opacity:0;transform:translateY(-100%);}to{opacity:1;transform:translateY(0);}"]);
var bodyHideAnimation = /*#__PURE__*/(0, _styledComponents.keyframes)(["from{opacity:1;transform:translateY(0);}to{opacity:0;transform:translateY(-100%);}"]);

var StyledWrapper = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web__sc-1d6y5ce-0"
})(["position:fixed;top:0;left:0;right:0;bottom:0;z-index:1;display:flex;align-items:center;justify-content:center;", ""], function (_ref) {
  var isHiding = _ref.isHiding;
  return isHiding && (0, _styledComponents.css)(["animation:0.1s ", " ease-out;"], wrapperHideAnimation);
});

var StyledOverlay = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web__sc-1d6y5ce-1"
})(["position:absolute;width:100%;height:100%;background-color:", ";cursor:pointer;"], function (_ref2) {
  var transparent = _ref2.transparent;
  return transparent ? 'transparent' : _plasmaCore.overlay;
});

var StyledModal = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web__sc-1d6y5ce-2"
})(["position:absolute;", ""], function (_ref3) {
  var isHiding = _ref3.isHiding;
  return (0, _styledComponents.css)(["animation:0.4s ", " ease-out;"], isHiding ? bodyHideAnimation : bodyShowAnimation);
});
/**
 * Модальное окно.
 * Управляет показом/скрытием, подложкой и анимацией визуальной части модального окна.
 */


var Modal = function Modal(_ref4) {
  var id = _ref4.id,
      isOpen = _ref4.isOpen,
      onClose = _ref4.onClose,
      rest = _objectWithoutProperties(_ref4, _excluded);

  var uniqId = (0, _plasmaCore.useUniqId)();
  var innerId = id || uniqId;

  var wrapperRef = _react["default"].useRef(null);

  var portalRef = _react["default"].useRef(null);

  var controller = _react["default"].useContext(_ModalsContext.ModalsContext);

  _react["default"].useEffect(function () {
    var portal = document.getElementById(_ModalsContext.MODALS_PORTAL_ID);

    if (!portal) {
      portal = document.createElement('div');
      portal.setAttribute('id', _ModalsContext.MODALS_PORTAL_ID);
      portal.setAttribute('aria-live', 'off');
      portal.setAttribute('role', 'presentation');
      portal.style.position = 'relative';
      portal.style.zIndex = '9000';
      document.body.appendChild(portal);
    }

    portalRef.current = portal;
    return function () {
      controller.unregister(innerId);
    };
  }, []);

  _react["default"].useEffect(function () {
    var onKeyDown = function onKeyDown(event) {
      if (event.keyCode === ESCAPE_KEYCODE && wrapperRef.current && portalRef.current && portalRef.current.contains(wrapperRef.current) && portalRef.current.children[portalRef.current.children.length - 1] === wrapperRef.current) {
        onClose === null || onClose === void 0 ? void 0 : onClose();
      }
    };

    window.addEventListener('keydown', onKeyDown);
    return function () {
      window.removeEventListener('keydown', onKeyDown);
    };
  }, []);

  if (isOpen) {
    controller.register(innerId);
  } else {
    controller.unregister(innerId);
    return null;
  }

  return portalRef.current && /*#__PURE__*/_reactDom["default"].createPortal( /*#__PURE__*/_react["default"].createElement(StyledWrapper, {
    ref: wrapperRef
  }, /*#__PURE__*/_react["default"].createElement(StyledOverlay, {
    transparent: controller.items.indexOf(innerId) !== 0,
    onClick: onClose
  }), /*#__PURE__*/_react["default"].createElement(StyledModal, null, /*#__PURE__*/_react["default"].createElement(_ModalView.ModalView, _extends({
    onClose: onClose
  }, rest)))), portalRef.current);
};

exports.Modal = Modal;
