// @sberdevices/plasma-tokens-web/themesValues/dark.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.speechBubbleReceived = exports.speechBubbleSent = exports.skeletonGradientLighter = exports.skeletonGradient = exports.voicePhraseGradient = exports.gradientDevice = exports.gradient = exports.buttonFocused = exports.buttonCheckedActiveColor = exports.buttonCheckedActive = exports.buttonCheckedHoverColor = exports.buttonCheckedHover = exports.buttonChecked = exports.buttonCriticalActive = exports.buttonCriticalHover = exports.buttonCritical = exports.buttonWarningActive = exports.buttonWarningHover = exports.buttonWarning = exports.buttonSuccessActive = exports.buttonSuccessHover = exports.buttonSuccess = exports.buttonAccent = exports.buttonSecondaryActive = exports.buttonSecondaryHover = exports.buttonSecondary = exports.buttonPrimaryActive = exports.buttonPrimaryHover = exports.buttonPrimary = exports.surfaceCard = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.surfaceSolid01 = exports.surfaceLiquid03 = exports.surfaceLiquid02 = exports.surfaceLiquid01 = exports.overlay = exports.critical = exports.warning = exports.success = exports.accent = exports.inputBorderFocus = exports.inputBorderHover = exports.inputBorder = exports.backgroundTertiary = exports.backgroundSecondary = exports.backgroundPrimary = exports.background = exports.inverse = exports.paragraph = exports.tertiary = exports.secondary = exports.primary = exports.linkVisitedActive = exports.linkVisitedHover = exports.linkVisited = exports.linkActive = exports.linkHover = exports.link = exports.text = exports.buttonWhiteSecondary = exports.buttonWhite = exports.buttonBlackTransparent = exports.buttonBlackSecondary = exports.buttonBlack = exports.buttonClear = exports.transparent = exports.dark03 = exports.dark02 = exports.dark01 = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.black = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.white = void 0;
/** Базовый белый цвет совпадает с whitePrimary */
exports.white = '#FFFFFF';
/** Первичный белый цвет */
exports.whitePrimary = '#FFFFFF';
/** Вторичный белый цвет */
exports.whiteSecondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный белый цвет */
exports.whiteTertiary = 'rgba(255, 255, 255, 0.28)';
/** Базовый черный цвет совпадает с blackPrimary */
exports.black = '#080808';
/** Первичный черный цвет */
exports.blackPrimary = '#080808';
/** Вторичный черныйцвет */
exports.blackSecondary = 'rgba(8, 8, 8, 0.56)';
/** Третичный черный цвет */
exports.blackTertiary = 'rgba(8, 8, 8, 0.28)';
/** Чёрный непрозрачный, не зависит от темы */
exports.dark01 = '#171717';
/** Чёрный непрозрачный, не зависит от темы */
exports.dark02 = '#232323';
/** Чёрный непрозрачный, не зависит от темы */
exports.dark03 = '#363636';
/** Прозрачный цвет */
exports.transparent = 'rgba(0, 0, 0, 0)';
/** Второстепенная прозрачная */
exports.buttonClear = 'rgba(0, 0, 0, 0)';
/** Дефолтная чёрная, не зависит от темы */
exports.buttonBlack = '#080808';
/** Второстепенная чёрная, не зависит от темы */
exports.buttonBlackSecondary = 'rgba(8, 8, 8, 0.12)';
/** Чёрная для использования поверх картинок */
exports.buttonBlackTransparent = 'rgba(8, 8, 8, 0.56)';
/** Дефолтная белая, не зависит от темы */
exports.buttonWhite = '#FFFFFF';
/** Второстепенная белая, не зависит от темы */
exports.buttonWhiteSecondary = 'rgba(255, 255, 255, 0.12)';
/** Базовый цвет текста, совпадает с primary */
exports.text = 'rgba(255, 255, 255, 0.96)';
exports.link = '#3F81FD';
exports.linkHover = '#6298FD';
exports.linkActive = '#1767FD';
exports.linkVisited = '#5993FF';
exports.linkVisitedHover = '#7DAAFF';
exports.linkVisitedActive = '#3078FF';
/** Основной текст в интерфейсе */
exports.primary = 'rgba(255, 255, 255, 0.96)';
/** Второстепенный текст */
exports.secondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный цвет текста */
exports.tertiary = 'rgba(255, 255, 255, 0.28)';
exports.paragraph = 'rgba(255, 255, 255, 0.96)';
exports.inverse = '#080808';
/** Основной бэкграунд всех приложений и смартапов */
exports.background = '#080808';
/** Первичный цвет фона */
exports.backgroundPrimary = '#171717';
/** Вторичный цвет фона */
exports.backgroundSecondary = '#232323';
/** Третичный цвет фона */
exports.backgroundTertiary = '#363636';
exports.inputBorder = 'rgba(255, 255, 255, 0.16)';
exports.inputBorderHover = 'rgba(255, 255, 255, 0.32)';
exports.inputBorderFocus = '#5993FF';
/** Акцентный цвет бренда */
exports.accent = '#5993FF';
/** Обозначение успешного сценария */
exports.success = '#24B23E';
/** Цвет предупреждения */
exports.warning = '#FA6D20';
/** Цвет ошибки */
exports.critical = '#FF4D5F';
/** Цвет фона паранжи */
exports.overlay = 'rgba(0, 0, 0, 0.8)';
/** Полупрозрачные поверхности */
exports.surfaceLiquid01 = 'rgba(255, 255, 255, 0.06)';
/** Полупрозрачные поверхности */
exports.surfaceLiquid02 = 'rgba(255, 255, 255, 0.12)';
/** Полупрозрачные поверхности */
exports.surfaceLiquid03 = 'rgba(255, 255, 255, 0.2)';
/** Непрозрачные поверхности */
exports.surfaceSolid01 = '#171717';
/** Непрозрачные поверхности */
exports.surfaceSolid02 = '#232323';
/** Непрозрачные поверхности */
exports.surfaceSolid03 = '#363636';
/** Основной фон для карточек */
exports.surfaceCard = 'rgba(255, 255, 255, 0.12)';
/** Первичный цвет контролов */
exports.buttonPrimary = '#3F82FD';
exports.buttonPrimaryHover = '#538efe';
exports.buttonPrimaryActive = '#3a75e5';
/** Второстепенная кнопка */
exports.buttonSecondary = 'rgba(255, 255, 255, 0.12)';
exports.buttonSecondaryHover = 'rgba(255, 255, 255, 0.16)';
exports.buttonSecondaryActive = 'rgba(255, 255, 255, 0.1)';
/** Акцентный цвет у контролов */
exports.buttonAccent = '#3F82FD';
/** Кнопка для успешного сценария */
exports.buttonSuccess = '#21A038';
exports.buttonSuccessHover = '#3eaa4d';
exports.buttonSuccessActive = '#1d9233';
/** Цвет предупреждения у контролов */
exports.buttonWarning = '#E35502';
exports.buttonWarningHover = '#f76d2f';
exports.buttonWarningActive = '#de550a';
/** Цвет ошибки у контролов */
exports.buttonCritical = '#FF2E43';
exports.buttonCriticalHover = '#ff4757';
exports.buttonCriticalActive = '#e82a3e';
/** Цвет зажатого контрола */
exports.buttonChecked = '#FFFFFF';
exports.buttonCheckedHover = '#FFFFFF';
exports.buttonCheckedHoverColor = 'hsla(0, 0%, 3.1372549019607843%, 1)';
exports.buttonCheckedActive = '#FFFFFF';
exports.buttonCheckedActiveColor = 'hsla(0, 0%, 3.1372549019607843%, 1)';
/** Цвет рамки фокуса у контрола */
exports.buttonFocused = '#3F82FD';
/** Градиент для заливки основного фона */
exports.gradient = 'linear-gradient(45deg, hsla(219.87577639751552, 100%, 68.43137254901961%, 1) 0%, hsla(190.1639344262295, 94.81865284974093%, 62.15686274509804%, 1) 100%)';
exports.gradientDevice = '';
exports.voicePhraseGradient = '';
exports.skeletonGradient = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% )';
exports.skeletonGradientLighter = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% )';
exports.speechBubbleSent = '';
exports.speechBubbleReceived = '';
