// @salutejs/plasma-tokens/brands/index.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.borderRadiusXxl = exports.borderRadiusXl = exports.borderRadiusL = exports.borderRadiusM = exports.borderRadiusS = exports.borderRadiusXs = exports.borderRadiusXxs = exports.shadowUpHardL = exports.shadowUpHardM = exports.shadowUpHardS = exports.shadowUpSoftL = exports.shadowUpSoftM = exports.shadowUpSoftS = exports.shadowDownHardL = exports.shadowDownHardM = exports.shadowDownHardS = exports.shadowDownSoftL = exports.shadowDownSoftM = exports.shadowDownSoftS = exports.buttonCritical = exports.buttonWarning = exports.buttonSuccess = exports.buttonAccent = exports.buttonSecondary = exports.surfaceCard = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.surfaceSolid01 = exports.surfaceLiquid03 = exports.surfaceLiquid02 = exports.surfaceLiquid01 = exports.overlay = exports.critical = exports.warning = exports.success = exports.accent = exports.background = exports.paragraph = exports.tertiary = exports.secondary = exports.primary = exports.text = exports.buttonWhiteSecondary = exports.buttonWhite = exports.buttonBlackSecondary = exports.buttonBlack = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.skeletonGradientLighter = exports.skeletonGradient = exports.inverseBackgroundTertiary = exports.inverseBackgroundSecondary = exports.inverseBackgroundPrimary = exports.lightBackgroundTertiary = exports.lightBackgroundSecondary = exports.lightBackgroundPrimary = exports.darkBackgroundTertiary = exports.darkBackgroundSecondary = exports.darkBackgroundPrimary = exports.backgroundTertiary = exports.backgroundSecondary = exports.backgroundPrimary = exports.inverseSurfaceTransparentWarning = exports.inverseSurfaceTransparentTertiary = exports.inverseSurfaceTransparentSecondary = exports.inverseSurfaceTransparentPrimary = exports.inverseSurfaceTransparentPositive = exports.inverseSurfaceTransparentNegative = exports.inverseSurfaceTransparentDeep = exports.inverseSurfaceTransparentCard = exports.inverseSurfaceTransparentAccentGradient = exports.inverseSurfaceTransparentAccent = exports.inverseSurfaceSolidCard = exports.inverseSurfaceSolidTertiary = exports.inverseSurfaceSolidSecondary = exports.inverseSurfaceSolidPrimary = exports.inverseSurfaceSolidDefault = exports.inverseSurfaceWarning = exports.inverseSurfacePositive = exports.inverseSurfaceNegative = exports.inverseSurfaceClear = exports.inverseSurfaceAccentGradient = exports.inverseSurfaceAccent = exports.onLightSurfaceTransparentWarning = exports.onLightSurfaceTransparentTertiary = exports.onLightSurfaceTransparentSecondary = exports.onLightSurfaceTransparentPrimary = exports.onLightSurfaceTransparentPositive = exports.onLightSurfaceTransparentNegative = exports.onLightSurfaceTransparentDeep = exports.onLightSurfaceTransparentCard = exports.onLightSurfaceTransparentAccentGradient = exports.onLightSurfaceTransparentAccent = exports.onLightSurfaceSolidCard = exports.onLightSurfaceSolidTertiary = exports.onLightSurfaceSolidSecondary = exports.onLightSurfaceSolidPrimary = exports.onLightSurfaceSolidDefault = exports.onLightSurfaceWarning = exports.onLightSurfacePositive = exports.onLightSurfaceNegative = exports.onLightSurfaceClear = exports.onLightSurfaceAccentGradient = exports.onLightSurfaceAccent = exports.onDarkSurfaceTransparentWarning = exports.onDarkSurfaceTransparentTertiary = exports.onDarkSurfaceTransparentSecondary = exports.onDarkSurfaceTransparentPrimary = exports.onDarkSurfaceTransparentPositive = exports.onDarkSurfaceTransparentNegative = exports.onDarkSurfaceTransparentDeep = exports.onDarkSurfaceTransparentCard = exports.onDarkSurfaceTransparentAccentGradient = exports.onDarkSurfaceTransparentAccent = exports.onDarkSurfaceSolidCard = exports.onDarkSurfaceSolidTertiary = exports.onDarkSurfaceSolidSecondary = exports.onDarkSurfaceSolidPrimary = exports.onDarkSurfaceSolidDefault = exports.onDarkSurfaceWarning = exports.onDarkSurfacePositive = exports.onDarkSurfaceNegative = exports.onDarkSurfaceClear = exports.onDarkSurfaceAccentGradient = exports.onDarkSurfaceAccent = exports.surfaceTransparentWarning = exports.surfaceTransparentTertiary = exports.surfaceTransparentSecondary = exports.surfaceTransparentPrimary = exports.surfaceTransparentPositive = exports.surfaceTransparentNegative = exports.surfaceTransparentDeep = exports.surfaceTransparentCard = exports.surfaceTransparentAccentGradient = exports.surfaceTransparentAccent = exports.surfaceSolidCard = exports.surfaceSolidTertiary = exports.surfaceSolidSecondary = exports.surfaceSolidPrimary = exports.surfaceSolidDefault = exports.surfaceWarning = exports.surfacePositive = exports.surfaceNegative = exports.surfaceClear = exports.surfaceAccentGradient = exports.surfaceAccent = exports.inverseTextTransparentWarning = exports.inverseTextTransparentPositive = exports.inverseTextTransparentNegative = exports.inverseTextTransparentAccentGradient = exports.inverseTextTransparentAccent = exports.inverseTextWarning = exports.inverseTextTertiary = exports.inverseTextSecondary = exports.inverseTextPrimary = exports.inverseTextPositive = exports.inverseTextParagraph = exports.inverseTextNegative = exports.inverseTextAccentGradient = exports.inverseTextAccent = exports.onLightTextTransparentWarning = exports.onLightTextTransparentPositive = exports.onLightTextTransparentNegative = exports.onLightTextTransparentAccentGradient = exports.onLightTextTransparentAccent = exports.onLightTextWarning = exports.onLightTextTertiary = exports.onLightTextSecondary = exports.onLightTextPrimary = exports.onLightTextPositive = exports.onLightTextParagraph = exports.onLightTextNegative = exports.onLightTextAccentGradient = exports.onLightTextAccent = exports.onDarkTextTransparentWarning = exports.onDarkTextTransparentPositive = exports.onDarkTextTransparentNegative = exports.onDarkTextTransparentAccentGradient = exports.onDarkTextTransparentAccent = exports.onDarkTextWarning = exports.onDarkTextTertiary = exports.onDarkTextSecondary = exports.onDarkTextPrimary = exports.onDarkTextPositive = exports.onDarkTextParagraph = exports.onDarkTextNegative = exports.onDarkTextAccentGradient = exports.onDarkTextAccent = exports.textTransparentWarning = exports.textTransparentPositive = exports.textTransparentNegative = exports.textTransparentAccentGradient = exports.textTransparentAccent = exports.textWarning = exports.textTertiary = exports.textSecondary = exports.textPrimary = exports.textPositive = exports.textParagraph = exports.textNegative = exports.textAccentGradient = exports.textAccent = exports.inverseOverlayHard = exports.inverseOverlaySoft = exports.onLightOverlayHard = exports.onLightOverlaySoft = exports.onDarkOverlayHard = exports.onDarkOverlaySoft = exports.overlayHard = exports.overlaySoft = void 0;
/** Цвет фона паранжи светлый */
exports.overlaySoft = 'var(--overlay-soft, rgba(0, 0, 0, 0.56))';
/** Цвет фона паранжи темный */
exports.overlayHard = 'var(--overlay-hard, rgba(0, 0, 0, 0.9))';
/** Цвет фона паранжи светлый на темном фоне */
exports.onDarkOverlaySoft = 'var(--on-dark-overlay-soft, rgba(0, 0, 0, 0.56))';
/** Цвет фона паранжи темный на темном фоне */
exports.onDarkOverlayHard = 'var(--on-dark-overlay-hard, rgba(0, 0, 0, 0.9))';
/** Цвет фона паранжи светлый на светлом фоне */
exports.onLightOverlaySoft = 'var(--on-light-overlay-soft, rgba(0, 0, 0, 0.56))';
/** Цвет фона паранжи темный на светлом фоне */
exports.onLightOverlayHard = 'var(--on-light-overlay-hard, rgba(0, 0, 0, 0.9))';
/** Инвертированный цвет фона паранжи светлый */
exports.inverseOverlaySoft = 'var(--inverse-overlay-soft, rgba(0, 0, 0, 0.56))';
/** Инвертированный цвет фона паранжи темный */
exports.inverseOverlayHard = 'var(--inverse-overlay-hard, rgba(0, 0, 0, 0.9))';
/** Акцентный цвет */
exports.textAccent = 'var(--text-accent, #FFFFFF)';
/** Акцентный цвет с градиентом */
exports.textAccentGradient = 'var(--text-accent-gradient, linear-gradient(135deg, black 0%, #FFFFFF 100%))';
/** Цвет ошибки */
exports.textNegative = 'var(--text-negative, #FF293E)';
/** Сплошной наборный текст */
exports.textParagraph = 'var(--text-paragraph, #ECECEC)';
/** Цвет успеха */
exports.textPositive = 'var(--text-positive, #1A9E32)';
/** Основной цвет текста */
exports.textPrimary = 'var(--text-primary, #F5F5F5)';
/** Вторичный цвет текста */
exports.textSecondary = 'var(--text-secondary, #B3B3B3)';
/** Третичный цвет текста */
exports.textTertiary = 'var(--text-tertiary, #7A7A7A)';
/** Цвет предупреждения */
exports.textWarning = 'var(--text-warning, #FA5F05)';
/** Прозрачный акцентный цвет */
exports.textTransparentAccent = 'var(--text-transparent-accent, rgba(255, 255, 255, 0.12))';
/** Прозрачный акцентный цвет с градиентом */
exports.textTransparentAccentGradient = 'var(--text-transparent-accent-gradient, linear-gradient(135deg, black 0%, rgba(255, 255, 255, 0.12) 100%))';
/** Прозрачный цвет ошибки */
exports.textTransparentNegative = 'var(--text-transparent-negative, rgba(255, 41, 62, 0.12))';
/** Прозрачный цвет успеха */
exports.textTransparentPositive = 'var(--text-transparent-positive, rgba(26, 158, 50, 0.12))';
/** Прозрачный цвет предупреждения */
exports.textTransparentWarning = 'var(--text-transparent-warning, rgba(250, 95, 5, 0.12))';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccent = 'var(--on-dark-text-accent, #FFFFFF)';
/** Акцентный цвет с градиентом на темном фоне */
exports.onDarkTextAccentGradient = 'var(--on-dark-text-accent-gradient, linear-gradient(135deg, black 0%, #FFFFFF 100%))';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegative = 'var(--on-dark-text-negative, #FF293E)';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraph = 'var(--on-dark-text-paragraph, #ECECEC)';
/** Цвет успеха на темном фоне */
exports.onDarkTextPositive = 'var(--on-dark-text-positive, #1A9E32)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimary = 'var(--on-dark-text-primary, #F5F5F5)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondary = 'var(--on-dark-text-secondary, #B3B3B3)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiary = 'var(--on-dark-text-tertiary, #7A7A7A)';
/** Цвет предупреждения на темном фоне */
exports.onDarkTextWarning = 'var(--on-dark-text-warning, #FA5F05)';
/** Прозрачный акцентный цвет на темном фоне */
exports.onDarkTextTransparentAccent = 'var(--on-dark-text-transparent-accent, rgba(255, 255, 255, 0.12))';
/** Прозрачный акцентный цвет с градиентом на темном фоне */
exports.onDarkTextTransparentAccentGradient = 'var(--on-dark-text-transparent-accent-gradient, linear-gradient(135deg, black 0%, rgba(255, 255, 255, 0.12) 100%))';
/** Прозрачный цвет ошибки на темном фоне */
exports.onDarkTextTransparentNegative = 'var(--on-dark-text-transparent-negative, rgba(255, 41, 62, 0.12))';
/** Прозрачный цвет успеха на темном фоне */
exports.onDarkTextTransparentPositive = 'var(--on-dark-text-transparent-positive, rgba(26, 158, 50, 0.12))';
/** Прозрачный цвет предупреждения на темном фоне */
exports.onDarkTextTransparentWarning = 'var(--on-dark-text-transparent-warning, rgba(250, 95, 5, 0.12))';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccent = 'var(--on-light-text-accent, #FFFFFF)';
/** Акцентный цвет с градиентом на светлом фоне */
exports.onLightTextAccentGradient = 'var(--on-light-text-accent-gradient, linear-gradient(135deg, white 0%, #FFFFFF 100%))';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegative = 'var(--on-light-text-negative, #F31B30)';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraph = 'var(--on-light-text-paragraph, #262626)';
/** Цвет успеха на светлом фоне */
exports.onLightTextPositive = 'var(--on-light-text-positive, #108E25)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimary = 'var(--on-light-text-primary, #171717)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondary = 'var(--on-light-text-secondary, #4E4E4E)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiary = 'var(--on-light-text-tertiary, #858585)';
/** Цвет предупреждения на светлом фоне */
exports.onLightTextWarning = 'var(--on-light-text-warning, #E85702)';
/** Прозрачный акцентный цвет на светлом фоне */
exports.onLightTextTransparentAccent = 'var(--on-light-text-transparent-accent, rgba(255, 255, 255, 0.12))';
/** Прозрачный акцентный цвет с градиентом на светлом фоне */
exports.onLightTextTransparentAccentGradient = 'var(--on-light-text-transparent-accent-gradient, linear-gradient(135deg, white 0%, rgba(255, 255, 255, 0.12) 100%))';
/** Прозрачный цвет ошибки на светлом фоне */
exports.onLightTextTransparentNegative = 'var(--on-light-text-transparent-negative, rgba(243, 27, 48, 0.12))';
/** Прозрачный цвет успеха на светлом фоне */
exports.onLightTextTransparentPositive = 'var(--on-light-text-transparent-positive, rgba(16, 142, 37, 0.12))';
/** Прозрачный цвет предупреждения на светлом фоне */
exports.onLightTextTransparentWarning = 'var(--on-light-text-transparent-warning, rgba(232, 87, 2, 0.12))';
/** Инвертированный акцентный цвет */
exports.inverseTextAccent = 'var(--inverse-text-accent, #FFFFFF)';
/** Инвертированный акцентный цвет с градиентом */
exports.inverseTextAccentGradient = 'var(--inverse-text-accent-gradient, linear-gradient(135deg, white 0%, #FFFFFF 100%))';
/** Инвертированный цвет ошибки */
exports.inverseTextNegative = 'var(--inverse-text-negative, #F31B30)';
/** Инвертированный cплошной наборный текст */
exports.inverseTextParagraph = 'var(--inverse-text-paragraph, #262626)';
/** Инвертированный цвет успеха */
exports.inverseTextPositive = 'var(--inverse-text-positive, #108E25)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimary = 'var(--inverse-text-primary, #171717)';
/** Инвертированный вторичный цвет текста */
exports.inverseTextSecondary = 'var(--inverse-text-secondary, #4E4E4E)';
/** Инвертированный третичный цвет текста */
exports.inverseTextTertiary = 'var(--inverse-text-tertiary, #858585)';
/** Инвертированный цвет предупреждения */
exports.inverseTextWarning = 'var(--inverse-text-warning, #E85702)';
/** Прозрачный инвертированный акцентный цвет */
exports.inverseTextTransparentAccent = 'var(--inverse-text-transparent-accent, rgba(255, 255, 255, 0.12))';
/** Прозрачный инвертированный акцентный цвет с градиентом */
exports.inverseTextTransparentAccentGradient = 'var(--inverse-text-transparent-accent-gradient, linear-gradient(135deg, white 0%, rgba(255, 255, 255, 0.12) 100%))';
/** Прозрачный инвертированный цвет ошибки */
exports.inverseTextTransparentNegative = 'var(--inverse-text-transparent-negative, rgba(243, 27, 48, 0.12))';
/** Прозрачный инвертированный цвет успеха */
exports.inverseTextTransparentPositive = 'var(--inverse-text-transparent-positive, rgba(16, 142, 37, 0.12))';
/** Прозрачный инвертированный цвет предупреждения */
exports.inverseTextTransparentWarning = 'var(--inverse-text-transparent-warning, rgba(232, 87, 2, 0.12))';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccent = 'var(--surface-accent, #FFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом */
exports.surfaceAccentGradient = 'var(--surface-accent-gradient, linear-gradient(135deg, black 0%, #FFFFFF 100%))';
/** Фон поверхности/контрола без заливки */
exports.surfaceClear = 'var(--surface-clear, rgba(0, 0, 0, 0))';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegative = 'var(--surface-negative, #FF293E)';
/** Цвет фона поверхности/контрола успех */
exports.surfacePositive = 'var(--surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола предупреждение */
exports.surfaceWarning = 'var(--surface-warning, #FA5F05)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
exports.surfaceSolidDefault = 'var(--surface-solid-default, #FFFFFF)';
/** Основной непрозрачный фон поверхности/контрола */
exports.surfaceSolidPrimary = 'var(--surface-solid-primary, #171717)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondary = 'var(--surface-solid-secondary, #262626)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiary = 'var(--surface-solid-tertiary, #363636)';
/** Основной фон для карточек */
exports.surfaceSolidCard = 'var(--surface-solid-card, #171717)';
/** Прозрачный акцентный фон поверхности/контрола */
exports.surfaceTransparentAccent = 'var(--surface-transparent-accent, rgba(255, 255, 255, 0.12))';
/** Прозрачный акцентный фон поверхности/контрола с градиентом */
exports.surfaceTransparentAccentGradient = 'var(--surface-transparent-accent-gradient, linear-gradient(135deg, black 0%, rgba(255, 255, 255, 0.12) 100%))';
/** Прозрачный фон для карточек */
exports.surfaceTransparentCard = 'var(--surface-transparent-card, rgba(255, 255, 255, 0.12))';
/** Непрозрачный фон поверхности/контрола */
exports.surfaceTransparentDeep = 'var(--surface-transparent-deep, rgba(255, 255, 255, 0.64))';
/** Прозрачный цвет фона поверхности/контрола ошибка */
exports.surfaceTransparentNegative = 'var(--surface-transparent-negative, rgba(255, 41, 62, 0.12))';
/** Прозрачный цвет фона поверхности/контрола успех */
exports.surfaceTransparentPositive = 'var(--surface-transparent-positive, rgba(26, 158, 50, 0.12))';
/** Основной прозрачный фон поверхности/контрола */
exports.surfaceTransparentPrimary = 'var(--surface-transparent-primary, rgba(255, 255, 255, 0.06))';
/** Вторичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentSecondary = 'var(--surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentTertiary = 'var(--surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
exports.surfaceTransparentWarning = 'var(--surface-transparent-warning, rgba(250, 95, 5, 0.12))';
/** Акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccent = 'var(--on-dark-surface-accent, #FFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
exports.onDarkSurfaceAccentGradient = 'var(--on-dark-surface-accent-gradient, linear-gradient(135deg, black 0%, #FFFFFF 100%))';
/** Фон поверхности/контрола без заливки на темном фоне */
exports.onDarkSurfaceClear = 'var(--on-dark-surface-clear, rgba(0, 0, 0, 0))';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegative = 'var(--on-dark-surface-negative, #FF293E)';
/** Цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfacePositive = 'var(--on-dark-surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола предупреждение на темном фоне */
exports.onDarkSurfaceWarning = 'var(--on-dark-surface-warning, #FA5F05)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
exports.onDarkSurfaceSolidDefault = 'var(--on-dark-surface-solid-default, #FFFFFF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidPrimary = 'var(--on-dark-surface-solid-primary, #171717)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #262626)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiary = 'var(--on-dark-surface-solid-tertiary, #363636)';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCard = 'var(--on-dark-surface-solid-card, #171717)';
/** Прозрачный акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentAccent = 'var(--on-dark-surface-transparent-accent, rgba(255, 255, 255, 0.12))';
/** Прозрачный акцентный фон поверхности/контрола с градиентом на темном фоне */
exports.onDarkSurfaceTransparentAccentGradient = 'var(--on-dark-surface-transparent-accent-gradient, linear-gradient(135deg, black 0%, rgba(255, 255, 255, 0.12) 100%))';
/** Прозрачный фон для карточек на темном фоне */
exports.onDarkSurfaceTransparentCard = 'var(--on-dark-surface-transparent-card, rgba(255, 255, 255, 0.12))';
/** Непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentDeep = 'var(--on-dark-surface-transparent-deep, rgba(255, 255, 255, 0.64))';
/** Прозрачный цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceTransparentNegative = 'var(--on-dark-surface-transparent-negative, rgba(255, 41, 62, 0.12))';
/** Прозрачный цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfaceTransparentPositive = 'var(--on-dark-surface-transparent-positive, rgba(26, 158, 50, 0.12))';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentPrimary = 'var(--on-dark-surface-transparent-primary, rgba(255, 255, 255, 0.06))';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentSecondary = 'var(--on-dark-surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentTertiary = 'var(--on-dark-surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный цвет фона поверхности/контрола предупреждение на темном фоне */
exports.onDarkSurfaceTransparentWarning = 'var(--on-dark-surface-transparent-warning, rgba(250, 95, 5, 0.12))';
/** Акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccent = 'var(--on-light-surface-accent, #FFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
exports.onLightSurfaceAccentGradient = 'var(--on-light-surface-accent-gradient, linear-gradient(135deg, white 0%, #FFFFFF 100%))';
/** Фон поверхности/контрола без заливки на светлом фоне */
exports.onLightSurfaceClear = 'var(--on-light-surface-clear, rgba(0, 0, 0, 0))';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegative = 'var(--on-light-surface-negative, #F31B30)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfacePositive = 'var(--on-light-surface-positive, #108E25)';
/** Цвет фона поверхности/контрола предупреждение на светлом фоне */
exports.onLightSurfaceWarning = 'var(--on-light-surface-warning, #E85702)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
exports.onLightSurfaceSolidDefault = 'var(--on-light-surface-solid-default, #080808)';
/** Основной непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidPrimary = 'var(--on-light-surface-solid-primary, #F5F5F5)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #ECECEC)';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidTertiary = 'var(--on-light-surface-solid-tertiary, #DDDDDD)';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCard = 'var(--on-light-surface-solid-card, #F7F7F7)';
/** Прозрачный акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentAccent = 'var(--on-light-surface-transparent-accent, rgba(255, 255, 255, 0.12))';
/** Прозрачный акцентный фон поверхности/контрола с градиентом на светлом фоне */
exports.onLightSurfaceTransparentAccentGradient = 'var(--on-light-surface-transparent-accent-gradient, linear-gradient(135deg, white 0%, rgba(255, 255, 255, 0.12) 100%))';
/** Прозрачный фон для карточек на светлом фоне */
exports.onLightSurfaceTransparentCard = 'var(--on-light-surface-transparent-card, rgba(8, 8, 8, 0.06))';
/** Непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentDeep = 'var(--on-light-surface-transparent-deep, rgba(0, 0, 0, 0.64))';
/** Прозрачный цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceTransparentNegative = 'var(--on-light-surface-transparent-negative, rgba(243, 27, 48, 0.12))';
/** Прозрачный цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfaceTransparentPositive = 'var(--on-light-surface-transparent-positive, rgba(16, 142, 37, 0.12))';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentPrimary = 'var(--on-light-surface-transparent-primary, rgba(8, 8, 8, 0.02))';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentSecondary = 'var(--on-light-surface-transparent-secondary, rgba(8, 8, 8, 0.06))';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentTertiary = 'var(--on-light-surface-transparent-tertiary, rgba(8, 8, 8, 0.12))';
/** Прозрачный цвет фона поверхности/контрола предупреждение на светлом фоне */
exports.onLightSurfaceTransparentWarning = 'var(--on-light-surface-transparent-warning, rgba(232, 87, 2, 0.12))';
/** Инвертированный акцентный фон поверхности/контрола */
exports.inverseSurfaceAccent = 'var(--inverse-surface-accent, #FFFFFF)';
/** Инвертированный акцентный фон поверхности/контрола с градиентом */
exports.inverseSurfaceAccentGradient = 'var(--inverse-surface-accent-gradient, linear-gradient(135deg, white 0%, #FFFFFF 100%))';
/** Инвертированный фон поверхности/контрола без заливки */
exports.inverseSurfaceClear = 'var(--inverse-surface-clear, rgba(0, 0, 0, 0))';
/** Инвертированный цвет фона поверхности/контрола ошибка */
exports.inverseSurfaceNegative = 'var(--inverse-surface-negative, #F31B30)';
/** Инвертированный цвет фона поверхности/контрола успех */
exports.inverseSurfacePositive = 'var(--inverse-surface-positive, #108E25)';
/** Инвертированный цвет фона поверхности/контрола предупреждение */
exports.inverseSurfaceWarning = 'var(--inverse-surface-warning, #E85702)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
exports.inverseSurfaceSolidDefault = 'var(--inverse-surface-solid-default, #080808)';
/** Инвертированный основной непрозрачный фон поверхности/контрола */
exports.inverseSurfaceSolidPrimary = 'var(--inverse-surface-solid-primary, #F5F5F5)';
/** Инвертированный вторичный непрозрачный фон поверхности/контрола */
exports.inverseSurfaceSolidSecondary = 'var(--inverse-surface-solid-secondary, #ECECEC)';
/** Инвертированный третичный непрозрачный фон поверхности/контрола */
exports.inverseSurfaceSolidTertiary = 'var(--inverse-surface-solid-tertiary, #DDDDDD)';
/** Инвертированный основной фон для карточек */
exports.inverseSurfaceSolidCard = 'var(--inverse-surface-solid-card, #F7F7F7)';
/** Прозрачный инвертированный акцентный фон поверхности/контрола */
exports.inverseSurfaceTransparentAccent = 'var(--inverse-surface-transparent-accent, rgba(255, 255, 255, 0.12))';
/** Прозрачный инвертированный акцентный фон поверхности/контрола с градиентом */
exports.inverseSurfaceTransparentAccentGradient = 'var(--inverse-surface-transparent-accent-gradient, linear-gradient(135deg, white 0%, rgba(255, 255, 255, 0.12) 100%))';
/** Инвертированный прозрачный фон для карточек */
exports.inverseSurfaceTransparentCard = 'var(--inverse-surface-transparent-card, rgba(8, 8, 8, 0.06))';
/** Инвертированный непрозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentDeep = 'var(--inverse-surface-transparent-deep, rgba(0, 0, 0, 0.64))';
/** Прозрачный инвертированный цвет фона поверхности/контрола ошибка */
exports.inverseSurfaceTransparentNegative = 'var(--inverse-surface-transparent-negative, rgba(243, 27, 48, 0.12))';
/** Прозрачный инвертированный цвет фона поверхности/контрола успех */
exports.inverseSurfaceTransparentPositive = 'var(--inverse-surface-transparent-positive, rgba(16, 142, 37, 0.12))';
/** Инвертированный основной прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentPrimary = 'var(--inverse-surface-transparent-primary, rgba(8, 8, 8, 0.02))';
/** Инвертированный вторичный прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentSecondary = 'var(--inverse-surface-transparent-secondary, rgba(8, 8, 8, 0.06))';
/** Инвертированный третичный прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentTertiary = 'var(--inverse-surface-transparent-tertiary, rgba(8, 8, 8, 0.12))';
/** Прозрачный инвертированный цвет фона поверхности/контрола предупреждение */
exports.inverseSurfaceTransparentWarning = 'var(--inverse-surface-transparent-warning, rgba(232, 87, 2, 0.12))';
/** Основной фон */
exports.backgroundPrimary = 'var(--background-primary, #000000)';
/** Вторичный фон */
exports.backgroundSecondary = 'var(--background-secondary, #080808)';
/** Третичный фон */
exports.backgroundTertiary = 'var(--background-tertiary, #171717)';
/** Основной фон на темном фоне */
exports.darkBackgroundPrimary = 'var(--dark-background-primary, #000000)';
/** Вторичный фон на темном фоне */
exports.darkBackgroundSecondary = 'var(--dark-background-secondary, #080808)';
/** Третичный фон на темном фоне */
exports.darkBackgroundTertiary = 'var(--dark-background-tertiary, #171717)';
/** Основной фон на светлом фоне */
exports.lightBackgroundPrimary = 'var(--light-background-primary, #FFFFFF)';
/** Вторичный фон на светлом фоне */
exports.lightBackgroundSecondary = 'var(--light-background-secondary, #FFFFFF)';
/** Третичный фон на светлом фоне */
exports.lightBackgroundTertiary = 'var(--light-background-tertiary, #FFFFFF)';
/** Инвертированный основной фон */
exports.inverseBackgroundPrimary = 'var(--inverse-background-primary, #FFFFFF)';
/** Инвертированный вторичный фон */
exports.inverseBackgroundSecondary = 'var(--inverse-background-secondary, #FFFFFF)';
/** Инвертированный третичный фон */
exports.inverseBackgroundTertiary = 'var(--inverse-background-tertiary, #FFFFFF)';
exports.skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
exports.skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** @deprecated instead use onDarkTextPrimary */
exports.whitePrimary = 'var(--plasma-colors-white-primary, var(--on-dark-text-primary))';
/** @deprecated instead use onDarkTextSecondary */
exports.whiteSecondary = 'var(--plasma-colors-white-secondary, var(--on-dark-text-secondary))';
/** @deprecated instead use onDarkTextTertiary */
exports.whiteTertiary = 'var(--plasma-colors-white-tertiary, var(--on-dark-text-tertiary))';
/** @deprecated instead use onLightTextPrimary */
exports.blackPrimary = 'var(--plasma-colors-black-primary, var(--on-light-text-primary))';
/** @deprecated instead use onLightTextSecondary */
exports.blackSecondary = 'var(--plasma-colors-black-secondary, var(--on-light-text-secondary))';
/** @deprecated instead use onLightTextTertiary */
exports.blackTertiary = 'var(--plasma-colors-black-tertiary, var(--on-light-text-tertiary))';
/** @deprecated instead use onLightSurfaceSolidDefault */
exports.buttonBlack = 'var(--plasma-colors-button-black, var(--on-light-surface-solid-default))';
/** @deprecated instead use onLightSurfaceTransparentSecondary */
exports.buttonBlackSecondary = 'var(--plasma-colors-button-black-secondary, var(--on-light-surface-transparent-secondary))';
/** @deprecated instead use onDarkSurfaceSolidDefault */
exports.buttonWhite = 'var(--plasma-colors-button-white, var(--on-dark-surface-solid-default))';
/** @deprecated instead use onDarkSurfaceTransparentSecondary */
exports.buttonWhiteSecondary = 'var(--plasma-colors-button-white-secondary, var(--on-dark-surface-transparent-secondary))';
/** @deprecated instead use textPrimary */
exports.text = 'var(--plasma-colors-text, var(--text-primary))';
/** @deprecated instead use textPrimary */
exports.primary = 'var(--plasma-colors-primary, var(--text-primary))';
/** @deprecated instead use textSecondary */
exports.secondary = 'var(--plasma-colors-secondary, var(--text-secondary))';
/** @deprecated instead use textTertiary */
exports.tertiary = 'var(--plasma-colors-tertiary, var(--text-tertiary))';
/** @deprecated instead use textParagraph */
exports.paragraph = 'var(--plasma-colors-paragraph, var(--text-paragraph))';
/** @deprecated instead use backgroundPrimary */
exports.background = 'var(--plasma-colors-background, var(--background-primary))';
/** @deprecated instead use textAccent */
exports.accent = 'var(--plasma-colors-accent, var(--text-accent))';
/** @deprecated instead use textPositive */
exports.success = 'var(--plasma-colors-success, var(--text-positive))';
/** @deprecated instead use textWarning */
exports.warning = 'var(--plasma-colors-warning, var(--text-warning))';
/** @deprecated instead use textNegative */
exports.critical = 'var(--plasma-colors-critical, var(--text-negative))';
/** @deprecated instead use overlaySoft */
exports.overlay = 'var(--plasma-colors-overlay, var(--overlay-soft))';
/** @deprecated instead use surfaceTransparentPrimary */
exports.surfaceLiquid01 = 'var(--plasma-colors-surface-liquid01, var(--surface-transparent-primary))';
/** @deprecated instead use surfaceTransparentSecondary */
exports.surfaceLiquid02 = 'var(--plasma-colors-surface-liquid02, var(--surface-transparent-secondary))';
/** @deprecated instead use surfaceTransparentTertiary */
exports.surfaceLiquid03 = 'var(--plasma-colors-surface-liquid03, var(--surface-transparent-tertiary))';
/** @deprecated instead use surfaceSolidPrimary */
exports.surfaceSolid01 = 'var(--plasma-colors-surface-solid01, var(--surface-solid-primary))';
/** @deprecated instead use surfaceSolidSecondary */
exports.surfaceSolid02 = 'var(--plasma-colors-surface-solid02, var(--surface-solid-secondary))';
/** @deprecated instead use surfaceSolidTertiary */
exports.surfaceSolid03 = 'var(--plasma-colors-surface-solid03, var(--surface-solid-tertiary))';
/** @deprecated instead use surfaceTransparentCard */
exports.surfaceCard = 'var(--plasma-colors-surface-card, var(--surface-transparent-card))';
/** @deprecated instead use surfaceTransparentSecondary */
exports.buttonSecondary = 'var(--plasma-colors-button-secondary, var(--surface-transparent-secondary))';
/** @deprecated instead use textAccent */
exports.buttonAccent = 'var(--plasma-colors-button-accent, var(--text-accent))';
/** @deprecated instead use surfacePositive */
exports.buttonSuccess = 'var(--plasma-colors-button-success, var(--surface-positive))';
/** @deprecated instead use surfaceWarning */
exports.buttonWarning = 'var(--plasma-colors-button-warning, var(--surface-warning))';
/** @deprecated instead use surfaceNegative */
exports.buttonCritical = 'var(--plasma-colors-button-critical, var(--surface-negative))';
exports.shadowDownSoftS = 'var(--shadow-down-soft-s, 0px 4px 14px -4px rgba(8, 8, 8, 0.08)), var(--shadow-down-soft-s, 0px 1px 4px -1px rgba(0, 0, 0, 0.04))';
exports.shadowDownSoftM = 'var(--shadow-down-soft-m, 0px 24px 48px -8px rgba(0, 0, 0, 0.08))';
exports.shadowDownSoftL = 'var(--shadow-down-soft-l, 0px 60px 112px -8px rgba(0, 0, 0, 0.08))';
exports.shadowDownHardS = 'var(--shadow-down-hard-s, 0px 4px 12px -3px rgba(8, 8, 8, 0.16)), var(--shadow-down-hard-s, 0px 1px 4px -2px rgba(0, 0, 0, 0.08))';
exports.shadowDownHardM = 'var(--shadow-down-hard-m, 0px 16px 32px -8px rgba(0, 0, 0, 0.24))';
exports.shadowDownHardL = 'var(--shadow-down-hard-l, 0px 60px 112px -8px rgba(0, 0, 0, 0.40))';
exports.shadowUpSoftS = 'var(--shadow-up-soft-s, 0px -4px 14px -4px rgba(8, 8, 8, 0.08)), var(--shadow-up-soft-s, 0px -1px 4px -1px rgba(0, 0, 0, 0.03))';
exports.shadowUpSoftM = 'var(--shadow-up-soft-m, 0px -24px 48px -8px rgba(0, 0, 0, 0.08))';
exports.shadowUpSoftL = 'var(--shadow-up-soft-l, 0px -60px 112px -8px rgba(0, 0, 0, 0.08))';
exports.shadowUpHardS = 'var(--shadow-up-hard-s, 0px -4px 12px -3px rgba(8, 8, 8, 0.20)), var(--shadow-up-hard-s, 0px -1px 4px -1px rgba(0, 0, 0, 0.03))';
exports.shadowUpHardM = 'var(--shadow-up-hard-m, 0px -16px 32px -8px rgba(0, 0, 0, 0.24))';
exports.shadowUpHardL = 'var(--shadow-up-hard-l, 0px -60px 112px -8px rgba(0, 0, 0, 0.40))';
exports.borderRadiusXxs = 'var(--border-radius-xxs, 4px)';
exports.borderRadiusXs = 'var(--border-radius-xs, 6px)';
exports.borderRadiusS = 'var(--border-radius-s, 8px)';
exports.borderRadiusM = 'var(--border-radius-m, 12px)';
exports.borderRadiusL = 'var(--border-radius-l, 16px)';
exports.borderRadiusXl = 'var(--border-radius-xl, 20px)';
exports.borderRadiusXxl = 'var(--border-radius-xxl, 40px)';
